[
  {
    "cmd": "delete-by-query",
    "query": {
      "contexts": [
        "package"
      ],
      "query": {
        "op": "case",
        "type": "word",
        "word": "BiobaseXNA"
      },
      "type": "context"
    }
  },
  [
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eThis module has the translation tables for the genetic code.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.AAseq",
          "name": "AAseq",
          "package": "BiobaseXNA",
          "source": "src/Biobase-AAseq.html",
          "type": "module"
        },
        "index": {
          "description": "This module has the translation tables for the genetic code",
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "AAseq",
          "package": "BiobaseXNA",
          "partial": "AAseq",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eThe amino acid newtype.\n\u003c/p\u003e",
          "module": "Biobase.AAseq",
          "name": "AA",
          "package": "BiobaseXNA",
          "source": "src/Biobase-AAseq.html#AA",
          "type": "newtype"
        },
        "index": {
          "description": "The amino acid newtype",
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "AA",
          "package": "BiobaseXNA",
          "partial": "AA",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#t:AA"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "AAseq",
          "package": "BiobaseXNA",
          "source": "src/Biobase-AAseq.html#AAseq",
          "type": "type"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "AAseq",
          "package": "BiobaseXNA",
          "partial": "AAseq",
          "type": "type"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#t:AAseq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eCreate amino acid sequences from different data sources.\n\u003c/p\u003e",
          "module": "Biobase.AAseq",
          "name": "MkAAseq",
          "package": "BiobaseXNA",
          "source": "src/Biobase-AAseq.html#MkAAseq",
          "type": "class"
        },
        "index": {
          "description": "Create amino acid sequences from different data sources",
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "MkAAseq",
          "package": "BiobaseXNA",
          "partial": "Mk AAseq",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#t:MkAAseq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "AA",
          "package": "BiobaseXNA",
          "signature": "AA",
          "source": "src/Biobase-AAseq.html#AA",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "AA",
          "package": "BiobaseXNA",
          "partial": "AA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:AA"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "aACharList",
          "package": "BiobaseXNA",
          "signature": "[(AA, Char)]",
          "source": "src/Biobase-AAseq.html#aACharList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "aACharList",
          "normalized": "[(AA,Char)]",
          "package": "BiobaseXNA",
          "partial": "AChar List",
          "signature": "[(AA,Char)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:aACharList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "aStop",
          "package": "BiobaseXNA",
          "signature": "AA",
          "source": "src/Biobase-AAseq.html#aStop",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "aStop",
          "package": "BiobaseXNA",
          "partial": "Stop",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:aStop"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "aaRange",
          "package": "BiobaseXNA",
          "signature": "[AA]",
          "source": "src/Biobase-AAseq.html#aaRange",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "aaRange",
          "normalized": "[AA]",
          "package": "BiobaseXNA",
          "partial": "Range",
          "signature": "[AA]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:aaRange"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "charAAList",
          "package": "BiobaseXNA",
          "signature": "[(Char, AA)]",
          "source": "src/Biobase-AAseq.html#charAAList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "charAAList",
          "normalized": "[(Char,AA)]",
          "package": "BiobaseXNA",
          "partial": "AAList",
          "signature": "[(Char,AA)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:charAAList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "codonTable",
          "package": "BiobaseXNA",
          "signature": "Map [Char] Char",
          "source": "src/Biobase-AAseq.html#codonTable",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "codonTable",
          "normalized": "Map[Char]Char",
          "package": "BiobaseXNA",
          "partial": "Table",
          "signature": "Map[Char]Char",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:codonTable"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eChar\u003c/a\u003e\u003c/code\u003e representation of an \u003ccode\u003e\u003ca\u003eAA\u003c/a\u003e\u003c/code\u003e.\n\u003c/p\u003e",
          "module": "Biobase.AAseq",
          "name": "fromAA",
          "package": "BiobaseXNA",
          "signature": "AA -\u003e Char",
          "source": "src/Biobase-AAseq.html#fromAA",
          "type": "function"
        },
        "index": {
          "description": "Char representation of an AA",
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "fromAA",
          "normalized": "AA-\u003eChar",
          "package": "BiobaseXNA",
          "partial": "AA",
          "signature": "AA-\u003eChar",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:fromAA"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "mkAAseq",
          "package": "BiobaseXNA",
          "signature": "x -\u003e Vector AA",
          "source": "src/Biobase-AAseq.html#mkAAseq",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "mkAAseq",
          "normalized": "a-\u003eVector AA",
          "package": "BiobaseXNA",
          "partial": "AAseq",
          "signature": "x-\u003eVector AA",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:mkAAseq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "nucCodonTable",
          "package": "BiobaseXNA",
          "signature": "Map [Nuc] AA",
          "source": "src/Biobase-AAseq.html#nucCodonTable",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "nucCodonTable",
          "normalized": "Map[Nuc]AA",
          "package": "BiobaseXNA",
          "partial": "Codon Table",
          "signature": "Map[Nuc]AA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:nucCodonTable"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eUsing the codon table, create an \u003ccode\u003e\u003ca\u003eAAseq\u003c/a\u003e\u003c/code\u003e from the \u003ccode\u003e\u003ca\u003ePrimary\u003c/a\u003e\u003c/code\u003e sequence.\n\u003c/p\u003e",
          "module": "Biobase.AAseq",
          "name": "primaryToAAseq",
          "package": "BiobaseXNA",
          "signature": "Primary -\u003e AAseq",
          "source": "src/Biobase-AAseq.html#primaryToAAseq",
          "type": "function"
        },
        "index": {
          "description": "Using the codon table create an AAseq from the Primary sequence",
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "primaryToAAseq",
          "normalized": "Primary-\u003eAAseq",
          "package": "BiobaseXNA",
          "partial": "To AAseq",
          "signature": "Primary-\u003eAAseq",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:primaryToAAseq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTranslate \u003ccode\u003e\u003ca\u003eChar\u003c/a\u003e\u003c/code\u003e amino acid representation into efficient \u003ccode\u003e\u003ca\u003eAA\u003c/a\u003e\u003c/code\u003e newtype.\n\u003c/p\u003e",
          "module": "Biobase.AAseq",
          "name": "toAA",
          "package": "BiobaseXNA",
          "signature": "Char -\u003e AA",
          "source": "src/Biobase-AAseq.html#toAA",
          "type": "function"
        },
        "index": {
          "description": "Translate Char amino acid representation into efficient AA newtype",
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "toAA",
          "normalized": "Char-\u003eAA",
          "package": "BiobaseXNA",
          "partial": "AA",
          "signature": "Char-\u003eAA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:toAA"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.AAseq",
          "name": "unAA",
          "package": "BiobaseXNA",
          "signature": "Int",
          "source": "src/Biobase-AAseq.html#AA",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase AAseq",
          "module": "Biobase.AAseq",
          "name": "unAA",
          "package": "BiobaseXNA",
          "partial": "AA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-AAseq.html#v:unAA"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eA special class of bounds for RNA/pair encodings that are used to index\n into tables. We typically encode more in the alphabets than we want to use\n to index, so in order to keep things simple, we have specialized bounds.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Primary.Bounds",
          "name": "Bounds",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary-Bounds.html",
          "type": "module"
        },
        "index": {
          "description": "special class of bounds for RNA pair encodings that are used to index into tables We typically encode more in the alphabets than we want to use to index so in order to keep things simple we have specialized bounds",
          "hierarchy": "Biobase Primary Bounds",
          "module": "Biobase.Primary.Bounds",
          "name": "Bounds",
          "package": "BiobaseXNA",
          "partial": "Bounds",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Bounds.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eminNormal\u003c/a\u003e\u003c/code\u003e and \u003ccode\u003e\u003ca\u003emaxNormal\u003c/a\u003e\u003c/code\u003e encode for, say, ACGU; while \u003ccode\u003e\u003ca\u003eminExtended\u003c/a\u003e\u003c/code\u003e and\n \u003ccode\u003e\u003ca\u003emaxExtended\u003c/a\u003e\u003c/code\u003e would allow \u003ccode\u003eN\u003c/code\u003e as well. See Biobase.RNA and\n Biobase.RNA.ViennaPair for instances.\n\u003c/p\u003e",
          "module": "Biobase.Primary.Bounds",
          "name": "Bounds",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary-Bounds.html#Bounds",
          "type": "class"
        },
        "index": {
          "description": "minNormal and maxNormal encode for say ACGU while minExtended and maxExtended would allow as well See Biobase.RNA and Biobase.RNA.ViennaPair for instances",
          "hierarchy": "Biobase Primary Bounds",
          "module": "Biobase.Primary.Bounds",
          "name": "Bounds",
          "package": "BiobaseXNA",
          "partial": "Bounds",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Bounds.html#t:Bounds"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.Bounds",
          "name": "maxExtended",
          "package": "BiobaseXNA",
          "signature": "a",
          "source": "src/Biobase-Primary-Bounds.html#maxExtended",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Primary Bounds",
          "module": "Biobase.Primary.Bounds",
          "name": "maxExtended",
          "package": "BiobaseXNA",
          "partial": "Extended",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Bounds.html#v:maxExtended"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.Bounds",
          "name": "maxNormal",
          "package": "BiobaseXNA",
          "signature": "a",
          "source": "src/Biobase-Primary-Bounds.html#maxNormal",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Primary Bounds",
          "module": "Biobase.Primary.Bounds",
          "name": "maxNormal",
          "package": "BiobaseXNA",
          "partial": "Normal",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Bounds.html#v:maxNormal"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.Bounds",
          "name": "minExtended",
          "package": "BiobaseXNA",
          "signature": "a",
          "source": "src/Biobase-Primary-Bounds.html#minExtended",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Primary Bounds",
          "module": "Biobase.Primary.Bounds",
          "name": "minExtended",
          "package": "BiobaseXNA",
          "partial": "Extended",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Bounds.html#v:minExtended"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.Bounds",
          "name": "minNormal",
          "package": "BiobaseXNA",
          "signature": "a",
          "source": "src/Biobase-Primary-Bounds.html#minNormal",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Primary Bounds",
          "module": "Biobase.Primary.Bounds",
          "name": "minNormal",
          "package": "BiobaseXNA",
          "partial": "Normal",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Bounds.html#v:minNormal"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eFast hash functions for \u003ccode\u003e\u003ca\u003ePrimary\u003c/a\u003e\u003c/code\u003e sequences. A hash is just an \u003ccode\u003e\u003ca\u003eInt\u003c/a\u003e\u003c/code\u003e, so\n use these only for short sequences.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Primary.Hashed",
          "name": "Hashed",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary-Hashed.html",
          "type": "module"
        },
        "index": {
          "description": "Fast hash functions for Primary sequences hash is just an Int so use these only for short sequences",
          "hierarchy": "Biobase Primary Hashed",
          "module": "Biobase.Primary.Hashed",
          "name": "Hashed",
          "package": "BiobaseXNA",
          "partial": "Hashed",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Hashed.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.Hashed",
          "name": "HashedPrimary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary-Hashed.html#HashedPrimary",
          "type": "newtype"
        },
        "index": {
          "hierarchy": "Biobase Primary Hashed",
          "module": "Biobase.Primary.Hashed",
          "name": "HashedPrimary",
          "package": "BiobaseXNA",
          "partial": "Hashed Primary",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Hashed.html#t:HashedPrimary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.Hashed",
          "name": "HashedPrimary",
          "package": "BiobaseXNA",
          "signature": "HashedPrimary Int",
          "source": "src/Biobase-Primary-Hashed.html#HashedPrimary",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary Hashed",
          "module": "Biobase.Primary.Hashed",
          "name": "HashedPrimary",
          "package": "BiobaseXNA",
          "partial": "Hashed Primary",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Hashed.html#v:HashedPrimary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eGiven a piece of primary sequence information, reduce it to an index.\n\u003c/p\u003e\u003cp\u003eWill throw an assertion in debug code if \u003ccode\u003eps\u003c/code\u003e are not within bounds. Note\n that \u003ca\u003emkPrimary [minBound]\u003c/a\u003e and \u003ca\u003emkPrimary [minBound,minBound]\u003c/a\u003e map to the\n same index. Meaning that indices are only unique within the same length\n group. Furthermore, indices with different (l,u)-bounds are not compatible\n with each other. All indices start at 0.\n\u003c/p\u003e\u003cp\u003eThe empty input produces an index of 0.\n\u003c/p\u003e\u003cp\u003eTODO currently goes the very inefficient way of creating a temporary vector\n for \u003ccode\u003eps\u003c/code\u003e. We could in O(1) create a vector from a Primary ...\n\u003c/p\u003e",
          "module": "Biobase.Primary.Hashed",
          "name": "mkHashedPrimary",
          "package": "BiobaseXNA",
          "signature": "(Nuc, Nuc) -\u003e Primary -\u003e HashedPrimary",
          "source": "src/Biobase-Primary-Hashed.html#mkHashedPrimary",
          "type": "function"
        },
        "index": {
          "description": "Given piece of primary sequence information reduce it to an index Will throw an assertion in debug code if ps are not within bounds Note that mkPrimary minBound and mkPrimary minBound minBound map to the same index Meaning that indices are only unique within the same length group Furthermore indices with different bounds are not compatible with each other All indices start at The empty input produces an index of TODO currently goes the very inefficient way of creating temporary vector for ps We could in create vector from Primary",
          "hierarchy": "Biobase Primary Hashed",
          "module": "Biobase.Primary.Hashed",
          "name": "mkHashedPrimary",
          "normalized": "(Nuc,Nuc)-\u003ePrimary-\u003eHashedPrimary",
          "package": "BiobaseXNA",
          "partial": "Hashed Primary",
          "signature": "(Nuc,Nuc)-\u003ePrimary-\u003eHashedPrimary",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-Hashed.html#v:mkHashedPrimary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eDegenerate base symbol representation. We use the same conventions as in\n \u003cimg src=\"https://en.wikipedia.org/wiki/Nucleic_acid_notation\"/\u003e which ignores\n \u003ccode\u003eU\u003c/code\u003eracil, except if it stands alone. Therefore, any RNA sequence should be\n converted to DNA (and back afterwards).\n\u003c/p\u003e\u003cp\u003eNOTE that the generic \u003ccode\u003e\u003ca\u003eChar\u003c/a\u003e\u003c/code\u003e instance is not optimized for speed.\n\u003c/p\u003e\u003cp\u003eTODO this should be easier once we have instances for RNA,DNA, etc\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Primary.IUPAC",
          "name": "IUPAC",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary-IUPAC.html",
          "type": "module"
        },
        "index": {
          "description": "Degenerate base symbol representation We use the same conventions as in which ignores racil except if it stands alone Therefore any RNA sequence should be converted to DNA and back afterwards NOTE that the generic Char instance is not optimized for speed TODO this should be easier once we have instances for RNA DNA etc",
          "hierarchy": "Biobase Primary IUPAC",
          "module": "Biobase.Primary.IUPAC",
          "name": "IUPAC",
          "package": "BiobaseXNA",
          "partial": "IUPAC",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-IUPAC.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.IUPAC",
          "name": "Degenerate",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary-IUPAC.html#Degenerate",
          "type": "class"
        },
        "index": {
          "hierarchy": "Biobase Primary IUPAC",
          "module": "Biobase.Primary.IUPAC",
          "name": "Degenerate",
          "package": "BiobaseXNA",
          "partial": "Degenerate",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-IUPAC.html#t:Degenerate"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.IUPAC",
          "name": "fromSymbol",
          "package": "BiobaseXNA",
          "signature": "x -\u003e [x]",
          "source": "src/Biobase-Primary-IUPAC.html#fromSymbol",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Primary IUPAC",
          "module": "Biobase.Primary.IUPAC",
          "name": "fromSymbol",
          "normalized": "a-\u003e[a]",
          "package": "BiobaseXNA",
          "partial": "Symbol",
          "signature": "x-\u003e[x]",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-IUPAC.html#v:fromSymbol"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003elist of characters\n\u003c/p\u003e",
          "module": "Biobase.Primary.IUPAC",
          "name": "iupacList",
          "package": "BiobaseXNA",
          "signature": "[(Char, String)]",
          "source": "src/Biobase-Primary-IUPAC.html#iupacList",
          "type": "function"
        },
        "index": {
          "description": "list of characters",
          "hierarchy": "Biobase Primary IUPAC",
          "module": "Biobase.Primary.IUPAC",
          "name": "iupacList",
          "normalized": "[(Char,String)]",
          "package": "BiobaseXNA",
          "partial": "List",
          "signature": "[(Char,String)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-IUPAC.html#v:iupacList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eRaw iupac data, embedded into the library.\n\u003c/p\u003e",
          "module": "Biobase.Primary.IUPAC",
          "name": "iupacNucleotides",
          "package": "BiobaseXNA",
          "signature": "ByteString",
          "source": "src/Biobase-Primary-IUPAC.html#iupacNucleotides",
          "type": "function"
        },
        "index": {
          "description": "Raw iupac data embedded into the library",
          "hierarchy": "Biobase Primary IUPAC",
          "module": "Biobase.Primary.IUPAC",
          "name": "iupacNucleotides",
          "package": "BiobaseXNA",
          "partial": "Nucleotides",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-IUPAC.html#v:iupacNucleotides"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary.IUPAC",
          "name": "toSymbol",
          "package": "BiobaseXNA",
          "signature": "[x] -\u003e Maybe x",
          "source": "src/Biobase-Primary-IUPAC.html#toSymbol",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Primary IUPAC",
          "module": "Biobase.Primary.IUPAC",
          "name": "toSymbol",
          "normalized": "[a]-\u003eMaybe a",
          "package": "BiobaseXNA",
          "partial": "Symbol",
          "signature": "[x]-\u003eMaybe x",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary-IUPAC.html#v:toSymbol"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eThe primary structure: interface to efficient encoding of RNA and DNA\n sequences. The design aims toward the \u003ccode\u003evector\u003c/code\u003e library and repa. In\n particular, everything is strict; if you want to stream full genomes, use\n \u003ccode\u003etext\u003c/code\u003e or lazy \u003ccode\u003ebytestring\u003c/code\u003es instead and cast to Biobase.Primary definitions\n only at the last moment.\n\u003c/p\u003e\u003cp\u003eNOTE individual nucleotides are encoded is \u003ccode\u003e\u003ca\u003eInt\u003c/a\u003e\u003c/code\u003es internally without any\n tagging. This means that we have no way of deciding if we are dealing with\n RNA or DNA on this level.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Primary",
          "name": "Primary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary.html",
          "type": "module"
        },
        "index": {
          "description": "The primary structure interface to efficient encoding of RNA and DNA sequences The design aims toward the vector library and repa In particular everything is strict if you want to stream full genomes use text or lazy bytestring instead and cast to Biobase.Primary definitions only at the last moment NOTE individual nucleotides are encoded is Int internally without any tagging This means that we have no way of deciding if we are dealing with RNA or DNA on this level",
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "Primary",
          "package": "BiobaseXNA",
          "partial": "Primary",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eGiven a sequence of nucleotides encoded in some \u003ca\u003etext-form\u003c/a\u003e, create a\n \u003ccode\u003e\u003ca\u003eNuc\u003c/a\u003e\u003c/code\u003e-based unboxed vector.\n\u003c/p\u003e",
          "module": "Biobase.Primary",
          "name": "MkPrimary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary.html#MkPrimary",
          "type": "class"
        },
        "index": {
          "description": "Given sequence of nucleotides encoded in some text-form create Nuc based unboxed vector",
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "MkPrimary",
          "package": "BiobaseXNA",
          "partial": "Mk Primary",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#t:MkPrimary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "Nuc",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary.html#Nuc",
          "type": "newtype"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "Nuc",
          "package": "BiobaseXNA",
          "partial": "Nuc",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#t:Nuc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "Primary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Primary.html#Primary",
          "type": "type"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "Primary",
          "package": "BiobaseXNA",
          "partial": "Primary",
          "type": "type"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#t:Primary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "Nuc",
          "package": "BiobaseXNA",
          "signature": "Nuc",
          "source": "src/Biobase-Primary.html#Nuc",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "Nuc",
          "package": "BiobaseXNA",
          "partial": "Nuc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:Nuc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "acgt",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "source": "src/Biobase-Primary.html#acgt",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "acgt",
          "normalized": "[Nuc]",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:acgt"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "acgu",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "source": "src/Biobase-Primary.html#acgu",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "acgu",
          "normalized": "[Nuc]",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:acgu"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "cgau",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "source": "src/Biobase-Primary.html#cgau",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "cgau",
          "normalized": "[Nuc]",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:cgau"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "charNucList",
          "package": "BiobaseXNA",
          "signature": "[(Char, Nuc)]",
          "source": "src/Biobase-Primary.html#charNucList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "charNucList",
          "normalized": "[(Char,Nuc)]",
          "package": "BiobaseXNA",
          "partial": "Nuc List",
          "signature": "[(Char,Nuc)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:charNucList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003eT\u003c/code\u003e to \u003ccode\u003eU\u003c/code\u003e\n\u003c/p\u003e",
          "module": "Biobase.Primary",
          "name": "convT2U",
          "package": "BiobaseXNA",
          "signature": "Nuc -\u003e Nuc",
          "source": "src/Biobase-Primary.html#convT2U",
          "type": "function"
        },
        "index": {
          "description": "to",
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "convT2U",
          "normalized": "Nuc-\u003eNuc",
          "package": "BiobaseXNA",
          "signature": "Nuc-\u003eNuc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:convT2U"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003eU\u003c/code\u003e to \u003ccode\u003eT\u003c/code\u003e\n\u003c/p\u003e",
          "module": "Biobase.Primary",
          "name": "convU2T",
          "package": "BiobaseXNA",
          "signature": "Nuc -\u003e Nuc",
          "source": "src/Biobase-Primary.html#convU2T",
          "type": "function"
        },
        "index": {
          "description": "to",
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "convU2T",
          "normalized": "Nuc-\u003eNuc",
          "package": "BiobaseXNA",
          "signature": "Nuc-\u003eNuc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:convU2T"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "fromNuc",
          "package": "BiobaseXNA",
          "signature": "Nuc -\u003e Char",
          "source": "src/Biobase-Primary.html#fromNuc",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "fromNuc",
          "normalized": "Nuc-\u003eChar",
          "package": "BiobaseXNA",
          "partial": "Nuc",
          "signature": "Nuc-\u003eChar",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:fromNuc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTranslate between \u003ccode\u003e\u003ca\u003eChar\u003c/a\u003e\u003c/code\u003es and \u003ccode\u003e\u003ca\u003eNuc\u003c/a\u003e\u003c/code\u003es.\n\u003c/p\u003e",
          "module": "Biobase.Primary",
          "name": "mkNuc",
          "package": "BiobaseXNA",
          "signature": "Char -\u003e Nuc",
          "source": "src/Biobase-Primary.html#mkNuc",
          "type": "function"
        },
        "index": {
          "description": "Translate between Char and Nuc",
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "mkNuc",
          "normalized": "Char-\u003eNuc",
          "package": "BiobaseXNA",
          "partial": "Nuc",
          "signature": "Char-\u003eNuc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:mkNuc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "mkPrimary",
          "package": "BiobaseXNA",
          "signature": "a -\u003e Primary",
          "source": "src/Biobase-Primary.html#mkPrimary",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "mkPrimary",
          "normalized": "a-\u003ePrimary",
          "package": "BiobaseXNA",
          "partial": "Primary",
          "signature": "a-\u003ePrimary",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:mkPrimary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "nN",
          "package": "BiobaseXNA",
          "signature": "Nuc",
          "source": "src/Biobase-Primary.html#nN",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "nN",
          "package": "BiobaseXNA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:nN"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "nacgt",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "source": "src/Biobase-Primary.html#nacgt",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "nacgt",
          "normalized": "[Nuc]",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:nacgt"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "nacgu",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "source": "src/Biobase-Primary.html#nacgu",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "nacgu",
          "normalized": "[Nuc]",
          "package": "BiobaseXNA",
          "signature": "[Nuc]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:nacgu"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "nucCharList",
          "package": "BiobaseXNA",
          "signature": "[(Nuc, Char)]",
          "source": "src/Biobase-Primary.html#nucCharList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "nucCharList",
          "normalized": "[(Nuc,Char)]",
          "package": "BiobaseXNA",
          "partial": "Char List",
          "signature": "[(Nuc,Char)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:nucCharList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Primary",
          "name": "unNuc",
          "package": "BiobaseXNA",
          "signature": "Int",
          "source": "src/Biobase-Primary.html#Nuc",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Primary",
          "module": "Biobase.Primary",
          "name": "unNuc",
          "package": "BiobaseXNA",
          "partial": "Nuc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Primary.html#v:unNuc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Constraint",
          "name": "Constraint",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Constraint.html",
          "type": "module"
        },
        "index": {
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "Constraint",
          "package": "BiobaseXNA",
          "partial": "Constraint",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eA constraint is nothing more than a vector of constraint characters\n together with a possible pairing for each character.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Constraint",
          "name": "Constraint",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Constraint.html#Constraint",
          "type": "newtype"
        },
        "index": {
          "description": "constraint is nothing more than vector of constraint characters together with possible pairing for each character",
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "Constraint",
          "package": "BiobaseXNA",
          "partial": "Constraint",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#t:Constraint"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eWe can create a constraint from different sources\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Constraint",
          "name": "MkConstraint",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Constraint.html#MkConstraint",
          "type": "class"
        },
        "index": {
          "description": "We can create constraint from different sources",
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "MkConstraint",
          "package": "BiobaseXNA",
          "partial": "Mk Constraint",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#t:MkConstraint"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Constraint",
          "name": "Constraint",
          "package": "BiobaseXNA",
          "signature": "Constraint",
          "source": "src/Biobase-Secondary-Constraint.html#Constraint",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "Constraint",
          "package": "BiobaseXNA",
          "partial": "Constraint",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#v:Constraint"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Constraint",
          "name": "bonusCC",
          "package": "BiobaseXNA",
          "signature": "Vector Char",
          "source": "src/Biobase-Secondary-Constraint.html#bonusCC",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "bonusCC",
          "package": "BiobaseXNA",
          "partial": "CC",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#v:bonusCC"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eGiven a \u003ccode\u003e\u003ca\u003eConstraint\u003c/a\u003e\u003c/code\u003e, create an NxN matrix with bonus energies. These\n energies can be included in all pair-creating functions and will disallow or\n strongly favor certain pairings, while others will receive neither bonus nor\n malus.\n\u003c/p\u003e\u003cp\u003eIn case, a pair (i,j) is annotated as both, bonus- and malus-receiving, it\n will be set to receive a malus. This can happen, if something like \u003ca\u003e\u003c\u003c/a\u003e would\n give a bonus, but \u003ca\u003ex\u003c/a\u003e gives a malus (and other cases).\n\u003c/p\u003e\u003cp\u003eTODO and again, we should parametrize over \u003ca\u003eEnergy\u003c/a\u003e, \u003ca\u003eScore\u003c/a\u003e, etc (that is,\n Prim a)\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Constraint",
          "name": "bonusTable",
          "package": "BiobaseXNA",
          "signature": "Double -\u003e Double -\u003e Constraint -\u003e Unboxed DIM2 Double",
          "source": "src/Biobase-Secondary-Constraint.html#bonusTable",
          "type": "function"
        },
        "index": {
          "description": "Given Constraint create an NxN matrix with bonus energies These energies can be included in all pair-creating functions and will disallow or strongly favor certain pairings while others will receive neither bonus nor malus In case pair is annotated as both bonus and malus-receiving it will be set to receive malus This can happen if something like would give bonus but gives malus and other cases TODO and again we should parametrize over Energy Score etc that is Prim",
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "bonusTable",
          "normalized": "Double-\u003eDouble-\u003eConstraint-\u003eUnboxed DIM Double",
          "package": "BiobaseXNA",
          "partial": "Table",
          "signature": "Double-\u003eDouble-\u003eConstraint-\u003eUnboxed DIM Double",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#v:bonusTable"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Constraint",
          "name": "mkConstraint",
          "package": "BiobaseXNA",
          "signature": "a -\u003e Constraint",
          "source": "src/Biobase-Secondary-Constraint.html#mkConstraint",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "mkConstraint",
          "normalized": "a-\u003eConstraint",
          "package": "BiobaseXNA",
          "partial": "Constraint",
          "signature": "a-\u003eConstraint",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#v:mkConstraint"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Constraint",
          "name": "nobonusCC",
          "package": "BiobaseXNA",
          "signature": "Vector Char",
          "source": "src/Biobase-Secondary-Constraint.html#nobonusCC",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "nobonusCC",
          "package": "BiobaseXNA",
          "partial": "CC",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#v:nobonusCC"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Constraint",
          "name": "unConstraint",
          "package": "BiobaseXNA",
          "signature": "Vector (Char, Int)",
          "source": "src/Biobase-Secondary-Constraint.html#Constraint",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Constraint",
          "module": "Biobase.Secondary.Constraint",
          "name": "unConstraint",
          "normalized": "Vector(Char,Int)",
          "package": "BiobaseXNA",
          "partial": "Constraint",
          "signature": "Vector(Char,Int)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Constraint.html#v:unConstraint"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eTypes for RNA secondary structure. Types vary from the simplest array\n (D1Secondary) to rather complex ones.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Secondary.Diagrams",
          "name": "Diagrams",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Diagrams.html",
          "type": "module"
        },
        "index": {
          "description": "Types for RNA secondary structure Types vary from the simplest array D1Secondary to rather complex ones",
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "Diagrams",
          "package": "BiobaseXNA",
          "partial": "Diagrams",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eRNA secondary structure with 1-diagrams. Each nucleotide is paired with at\n most one other nucleotide. A nucleotide with index k in [0..len-1] is\n paired if unD1S VU.! k \u003e 0.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Diagrams",
          "name": "D1Secondary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Diagrams.html#D1Secondary",
          "type": "newtype"
        },
        "index": {
          "description": "RNA secondary structure with diagrams Each nucleotide is paired with at most one other nucleotide nucleotide with index in len-1 is paired if unD1S VU",
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "D1Secondary",
          "package": "BiobaseXNA",
          "partial": "Secondary",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#t:D1Secondary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "D2Secondary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Diagrams.html#D2Secondary",
          "type": "newtype"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "D2Secondary",
          "package": "BiobaseXNA",
          "partial": "Secondary",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#t:D2Secondary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "MkD1Secondary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Diagrams.html#MkD1Secondary",
          "type": "class"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "MkD1Secondary",
          "package": "BiobaseXNA",
          "partial": "Mk Secondary",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#t:MkD1Secondary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "MkD2Secondary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Diagrams.html#MkD2Secondary",
          "type": "class"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "MkD2Secondary",
          "package": "BiobaseXNA",
          "partial": "Mk Secondary",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#t:MkD2Secondary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eA secondary-structure tree. Has no notion of pseudoknots.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Diagrams",
          "name": "SSTree",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Diagrams.html#SSTree",
          "type": "data"
        },
        "index": {
          "description": "secondary-structure tree Has no notion of pseudoknots",
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "SSTree",
          "package": "BiobaseXNA",
          "partial": "SSTree",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#t:SSTree"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "D1S",
          "package": "BiobaseXNA",
          "signature": "D1S",
          "source": "src/Biobase-Secondary-Diagrams.html#D1Secondary",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "D1S",
          "package": "BiobaseXNA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:D1S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "D2S",
          "package": "BiobaseXNA",
          "signature": "D2S",
          "source": "src/Biobase-Secondary-Diagrams.html#D2Secondary",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "D2S",
          "package": "BiobaseXNA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:D2S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "SSExt",
          "package": "BiobaseXNA",
          "signature": "SSExt Int a [SSTree idx a]",
          "source": "src/Biobase-Secondary-Diagrams.html#SSTree",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "SSExt",
          "normalized": "SSExt Int a[SSTree b a]",
          "package": "BiobaseXNA",
          "partial": "SSExt",
          "signature": "SSExt Int a[SSTree idx a]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:SSExt"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "SSTree",
          "package": "BiobaseXNA",
          "signature": "SSTree idx a [SSTree idx a]",
          "source": "src/Biobase-Secondary-Diagrams.html#SSTree",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "SSTree",
          "normalized": "SSTree a b[SSTree a b]",
          "package": "BiobaseXNA",
          "partial": "SSTree",
          "signature": "SSTree idx a[SSTree idx a]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:SSTree"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eCreate a tree from (pseudoknot-free [not checked]) 1-diagrams.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Diagrams",
          "name": "d1sTree",
          "package": "BiobaseXNA",
          "signature": "D1Secondary -\u003e SSTree PairIdx ()",
          "source": "src/Biobase-Secondary-Diagrams.html#d1sTree",
          "type": "function"
        },
        "index": {
          "description": "Create tree from pseudoknot-free not checked diagrams",
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "d1sTree",
          "normalized": "D Secondary-\u003eSSTree PairIdx()",
          "package": "BiobaseXNA",
          "partial": "Tree",
          "signature": "D Secondary-\u003eSSTree PairIdx()",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:d1sTree"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "d2Compare",
          "package": "BiobaseXNA",
          "signature": "((t2, t3), t) -\u003e ((t2, t3), t1) -\u003e Ordering",
          "source": "src/Biobase-Secondary-Diagrams.html#d2Compare",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "d2Compare",
          "normalized": "((a,a),a)-\u003e((a,a),a)-\u003eOrdering",
          "package": "BiobaseXNA",
          "partial": "Compare",
          "signature": "((t,t),t)-\u003e((t,t),t)-\u003eOrdering",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:d2Compare"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "d2Grouping",
          "package": "BiobaseXNA",
          "signature": "((a, a1), t) -\u003e ((a, a1), t1) -\u003e Bool",
          "source": "src/Biobase-Secondary-Diagrams.html#d2Grouping",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "d2Grouping",
          "normalized": "((a,a),b)-\u003e((a,a),b)-\u003eBool",
          "package": "BiobaseXNA",
          "partial": "Grouping",
          "signature": "((a,a),t)-\u003e((a,a),t)-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:d2Grouping"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eCreate a tree from (pseudoknot-free [not checked]) 2-diagrams.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Diagrams",
          "name": "d2sTree",
          "package": "BiobaseXNA",
          "signature": "D2Secondary -\u003e SSTree ExtPairIdx ()",
          "source": "src/Biobase-Secondary-Diagrams.html#d2sTree",
          "type": "function"
        },
        "index": {
          "description": "Create tree from pseudoknot-free not checked diagrams",
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "d2sTree",
          "normalized": "D Secondary-\u003eSSTree ExtPairIdx()",
          "package": "BiobaseXNA",
          "partial": "Tree",
          "signature": "D Secondary-\u003eSSTree ExtPairIdx()",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:d2sTree"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eSecondary structure parser which allows pseudoknots, if they use different\n kinds of brackets.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Diagrams",
          "name": "dotBracket",
          "package": "BiobaseXNA",
          "signature": "[String] -\u003e String -\u003e [(Int, Int)]",
          "source": "src/Biobase-Secondary-Diagrams.html#dotBracket",
          "type": "function"
        },
        "index": {
          "description": "Secondary structure parser which allows pseudoknots if they use different kinds of brackets",
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "dotBracket",
          "normalized": "[String]-\u003eString-\u003e[(Int,Int)]",
          "package": "BiobaseXNA",
          "partial": "Bracket",
          "signature": "[String]-\u003eString-\u003e[(Int,Int)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:dotBracket"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "fromD1S",
          "package": "BiobaseXNA",
          "signature": "D1Secondary -\u003e a",
          "source": "src/Biobase-Secondary-Diagrams.html#fromD1S",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "fromD1S",
          "normalized": "D Secondary-\u003ea",
          "package": "BiobaseXNA",
          "signature": "D Secondary-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:fromD1S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "fromD2S",
          "package": "BiobaseXNA",
          "signature": "D2Secondary -\u003e a",
          "source": "src/Biobase-Secondary-Diagrams.html#fromD2S",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "fromD2S",
          "normalized": "D Secondary-\u003ea",
          "package": "BiobaseXNA",
          "signature": "D Secondary-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:fromD2S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "mkD1S",
          "package": "BiobaseXNA",
          "signature": "a -\u003e D1Secondary",
          "source": "src/Biobase-Secondary-Diagrams.html#mkD1S",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "mkD1S",
          "normalized": "a-\u003eD Secondary",
          "package": "BiobaseXNA",
          "signature": "a-\u003eD Secondary",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:mkD1S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "mkD2S",
          "package": "BiobaseXNA",
          "signature": "a -\u003e D2Secondary",
          "source": "src/Biobase-Secondary-Diagrams.html#mkD2S",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "mkD2S",
          "normalized": "a-\u003eD Secondary",
          "package": "BiobaseXNA",
          "signature": "a-\u003eD Secondary",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:mkD2S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "unD1S",
          "package": "BiobaseXNA",
          "signature": "Vector Int",
          "source": "src/Biobase-Secondary-Diagrams.html#D1Secondary",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "unD1S",
          "package": "BiobaseXNA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:unD1S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Diagrams",
          "name": "unD2S",
          "package": "BiobaseXNA",
          "signature": "Vector ((Int, Edge, CTisomerism), (Int, Edge, CTisomerism))",
          "source": "src/Biobase-Secondary-Diagrams.html#D2Secondary",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Diagrams",
          "module": "Biobase.Secondary.Diagrams",
          "name": "unD2S",
          "normalized": "Vector((Int,Edge,CTisomerism),(Int,Edge,CTisomerism))",
          "package": "BiobaseXNA",
          "signature": "Vector((Int,Edge,CTisomerism),(Int,Edge,CTisomerism))",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Diagrams.html#v:unD2S"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eProvides detailed information on isostericity of RNA basepairs. All data\n is extracted from csv files which were created from supplemental files in:\n\u003c/p\u003e\u003cp\u003eFrequency and isostericity of RNA base pairs\n Jesse Stombaugh, Craig L. Zirbel, Eric Westhof, and Neocles B. Leontis\n Nucl. Acids Res. (2009)\n doi:10.1093\u003cem\u003enar\u003c/em\u003egkp011\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "Isostericity",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Isostericity.html",
          "type": "module"
        },
        "index": {
          "description": "Provides detailed information on isostericity of RNA basepairs All data is extracted from csv files which were created from supplemental files in Frequency and isostericity of RNA base pairs Jesse Stombaugh Craig Zirbel Eric Westhof and Neocles Leontis Nucl Acids Res doi nar gkp011",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "Isostericity",
          "package": "BiobaseXNA",
          "partial": "Isostericity",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eMethods to determine the isostericity classes for a given basepair type,\n or alternatively which basepair types are in a certain isostericity class.\n\u003c/p\u003e\u003cp\u003eTODO This requires a major cleanup: right now we are handling \u003ccode\u003e\u003ca\u003eString\u003c/a\u003e\u003c/code\u003es as\n class descriptors, but we should really be newtype-wrapping or create enum\n data constructors.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "IsostericityLookup",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Isostericity.html#IsostericityLookup",
          "type": "class"
        },
        "index": {
          "description": "Methods to determine the isostericity classes for given basepair type or alternatively which basepair types are in certain isostericity class TODO This requires major cleanup right now we are handling String as class descriptors but we should really be newtype-wrapping or create enum data constructors",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "IsostericityLookup",
          "package": "BiobaseXNA",
          "partial": "Isostericity Lookup",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#t:IsostericityLookup"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eThe default isostericity mapping.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "defaultIsostericityMap",
          "package": "BiobaseXNA",
          "signature": "Map ExtPair [String]",
          "source": "src/Biobase-Secondary-Isostericity.html#defaultIsostericityMap",
          "type": "function"
        },
        "index": {
          "description": "The default isostericity mapping",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "defaultIsostericityMap",
          "normalized": "Map ExtPair[String]",
          "package": "BiobaseXNA",
          "partial": "Isostericity Map",
          "signature": "Map ExtPair[String]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#v:defaultIsostericityMap"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eRaw CSV data, embedded into the library.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "detailedCSV",
          "package": "BiobaseXNA",
          "signature": "ByteString",
          "source": "src/Biobase-Secondary-Isostericity.html#detailedCSV",
          "type": "function"
        },
        "index": {
          "description": "Raw CSV data embedded into the library",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "detailedCSV",
          "package": "BiobaseXNA",
          "partial": "CSV",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#v:detailedCSV"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTo which classes does a basepair+type belong\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "getClasses",
          "package": "BiobaseXNA",
          "signature": "a -\u003e [String]",
          "source": "src/Biobase-Secondary-Isostericity.html#getClasses",
          "type": "method"
        },
        "index": {
          "description": "To which classes does basepair type belong",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "getClasses",
          "normalized": "a-\u003e[String]",
          "package": "BiobaseXNA",
          "partial": "Classes",
          "signature": "a-\u003e[String]",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#v:getClasses"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eWhat basepairs+type are in a particular class\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "inClass",
          "package": "BiobaseXNA",
          "signature": "String -\u003e [a]",
          "source": "src/Biobase-Secondary-Isostericity.html#inClass",
          "type": "method"
        },
        "index": {
          "description": "What basepairs type are in particular class",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "inClass",
          "normalized": "String-\u003e[a]",
          "package": "BiobaseXNA",
          "partial": "Class",
          "signature": "String-\u003e[a]",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#v:inClass"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eProcess CSV list-of-lists to get the isostericity data.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "mkIsostericityList",
          "package": "BiobaseXNA",
          "signature": "[[[String]]] -\u003e [(ExtPair, [String])]",
          "source": "src/Biobase-Secondary-Isostericity.html#mkIsostericityList",
          "type": "function"
        },
        "index": {
          "description": "Process CSV list-of-lists to get the isostericity data",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "mkIsostericityList",
          "normalized": "[[[String]]]-\u003e[(ExtPair,[String])]",
          "package": "BiobaseXNA",
          "partial": "Isostericity List",
          "signature": "[[[String]]]-\u003e[(ExtPair,[String])]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#v:mkIsostericityList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eMapping of (pair,pairtype) to isostericity class.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "mkIsostericityMap",
          "package": "BiobaseXNA",
          "signature": "[[[String]]] -\u003e Map ExtPair [String]",
          "source": "src/Biobase-Secondary-Isostericity.html#mkIsostericityMap",
          "type": "function"
        },
        "index": {
          "description": "Mapping of pair pairtype to isostericity class",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "mkIsostericityMap",
          "normalized": "[[[String]]]-\u003eMap ExtPair[String]",
          "package": "BiobaseXNA",
          "partial": "Isostericity Map",
          "signature": "[[[String]]]-\u003eMap ExtPair[String]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#v:mkIsostericityMap"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eSimple parsing of raw CSV data.\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Isostericity",
          "name": "parsedCSV",
          "package": "BiobaseXNA",
          "signature": "[[[[Char]]]]",
          "source": "src/Biobase-Secondary-Isostericity.html#parsedCSV",
          "type": "function"
        },
        "index": {
          "description": "Simple parsing of raw CSV data",
          "hierarchy": "Biobase Secondary Isostericity",
          "module": "Biobase.Secondary.Isostericity",
          "name": "parsedCSV",
          "normalized": "[[[[Char]]]]",
          "package": "BiobaseXNA",
          "partial": "CSV",
          "signature": "[[[[Char]]]]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Isostericity.html#v:parsedCSV"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eMethods to transform a secondary structure containing pseudoknots into a\n structure which is pseudoknot-free.\n\u003c/p\u003e\u003cp\u003eTODO Until a better name is found, this module is home to functions for\n \u003ca\u003ede-pseudoknotting\u003c/a\u003e structures.\n\u003c/p\u003e\u003cp\u003eTODO Check if there are corner-cases remaining when considering 2-diagrams.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Secondary.PseudoKnots",
          "name": "PseudoKnots",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-PseudoKnots.html",
          "type": "module"
        },
        "index": {
          "description": "Methods to transform secondary structure containing pseudoknots into structure which is pseudoknot-free TODO Until better name is found this module is home to functions for de-pseudoknotting structures TODO Check if there are corner-cases remaining when considering diagrams",
          "hierarchy": "Biobase Secondary PseudoKnots",
          "module": "Biobase.Secondary.PseudoKnots",
          "name": "PseudoKnots",
          "package": "BiobaseXNA",
          "partial": "Pseudo Knots",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-PseudoKnots.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTry to removed pseudoknots from the \u003ca\u003epairlist\u003c/a\u003e. This works by counting for\n each pair, how many pairs are incompatible with it. Then those with most\n incompatibilities are successively removed. This function might well remove\n more than necessary!\n\u003c/p\u003e",
          "module": "Biobase.Secondary.PseudoKnots",
          "name": "RemovePseudoKnots",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-PseudoKnots.html#RemovePseudoKnots",
          "type": "class"
        },
        "index": {
          "description": "Try to removed pseudoknots from the pairlist This works by counting for each pair how many pairs are incompatible with it Then those with most incompatibilities are successively removed This function might well remove more than necessary",
          "hierarchy": "Biobase Secondary PseudoKnots",
          "module": "Biobase.Secondary.PseudoKnots",
          "name": "RemovePseudoKnots",
          "package": "BiobaseXNA",
          "partial": "Remove Pseudo Knots",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-PseudoKnots.html#t:RemovePseudoKnots"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.PseudoKnots",
          "name": "removeByCounting",
          "package": "BiobaseXNA",
          "signature": "a -\u003e a",
          "source": "src/Biobase-Secondary-PseudoKnots.html#removeByCounting",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary PseudoKnots",
          "module": "Biobase.Secondary.PseudoKnots",
          "name": "removeByCounting",
          "normalized": "a-\u003ea",
          "package": "BiobaseXNA",
          "partial": "By Counting",
          "signature": "a-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-PseudoKnots.html#v:removeByCounting"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eEncoding of Watson-Crick and Wobble Pairs in the Vienna RNA package style.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Secondary.Vienna",
          "name": "Vienna",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Vienna.html",
          "type": "module"
        },
        "index": {
          "description": "Encoding of Watson-Crick and Wobble Pairs in the Vienna RNA package style",
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "Vienna",
          "package": "BiobaseXNA",
          "partial": "Vienna",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "MkViennaPair",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Vienna.html#MkViennaPair",
          "type": "class"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "MkViennaPair",
          "package": "BiobaseXNA",
          "partial": "Mk Vienna Pair",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#t:MkViennaPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eUse machine Ints internally\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Vienna",
          "name": "ViennaPair",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary-Vienna.html#ViennaPair",
          "type": "newtype"
        },
        "index": {
          "description": "Use machine Ints internally",
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "ViennaPair",
          "package": "BiobaseXNA",
          "partial": "Vienna Pair",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#t:ViennaPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "ViennaPair",
          "package": "BiobaseXNA",
          "signature": "ViennaPair Int",
          "source": "src/Biobase-Secondary-Vienna.html#ViennaPair",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "ViennaPair",
          "package": "BiobaseXNA",
          "partial": "Vienna Pair",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:ViennaPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "cgnsP",
          "package": "BiobaseXNA",
          "signature": "[ViennaPair]",
          "source": "src/Biobase-Secondary-Vienna.html#cgnsP",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "cgnsP",
          "normalized": "[ViennaPair]",
          "package": "BiobaseXNA",
          "signature": "[ViennaPair]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:cgnsP"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "cguaP",
          "package": "BiobaseXNA",
          "signature": "[ViennaPair]",
          "source": "src/Biobase-Secondary-Vienna.html#cguaP",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "cguaP",
          "normalized": "[ViennaPair]",
          "package": "BiobaseXNA",
          "signature": "[ViennaPair]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:cguaP"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "fromViennaPair",
          "package": "BiobaseXNA",
          "signature": "ViennaPair -\u003e a",
          "source": "src/Biobase-Secondary-Vienna.html#fromViennaPair",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "fromViennaPair",
          "normalized": "ViennaPair-\u003ea",
          "package": "BiobaseXNA",
          "partial": "Vienna Pair",
          "signature": "ViennaPair-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:fromViennaPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "isViennaPair",
          "package": "BiobaseXNA",
          "signature": "Nuc -\u003e Nuc -\u003e Bool",
          "source": "src/Biobase-Secondary-Vienna.html#isViennaPair",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "isViennaPair",
          "normalized": "Nuc-\u003eNuc-\u003eBool",
          "package": "BiobaseXNA",
          "partial": "Vienna Pair",
          "signature": "Nuc-\u003eNuc-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:isViennaPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "mkViennaPair",
          "package": "BiobaseXNA",
          "signature": "a -\u003e ViennaPair",
          "source": "src/Biobase-Secondary-Vienna.html#mkViennaPair",
          "type": "method"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "mkViennaPair",
          "normalized": "a-\u003eViennaPair",
          "package": "BiobaseXNA",
          "partial": "Vienna Pair",
          "signature": "a-\u003eViennaPair",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:mkViennaPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "pairToString",
          "package": "BiobaseXNA",
          "signature": "[(ViennaPair, [Char])]",
          "source": "src/Biobase-Secondary-Vienna.html#pairToString",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "pairToString",
          "normalized": "[(ViennaPair,[Char])]",
          "package": "BiobaseXNA",
          "partial": "To String",
          "signature": "[(ViennaPair,[Char])]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:pairToString"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003ereverse a vienna pair\n\u003c/p\u003e",
          "module": "Biobase.Secondary.Vienna",
          "name": "revPair",
          "package": "BiobaseXNA",
          "signature": "ViennaPair -\u003e ViennaPair",
          "source": "src/Biobase-Secondary-Vienna.html#revPair",
          "type": "function"
        },
        "index": {
          "description": "reverse vienna pair",
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "revPair",
          "normalized": "ViennaPair-\u003eViennaPair",
          "package": "BiobaseXNA",
          "partial": "Pair",
          "signature": "ViennaPair-\u003eViennaPair",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:revPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "viennaPairTable",
          "package": "BiobaseXNA",
          "signature": "Unboxed ((Z :. Nuc) :. Nuc) ViennaPair",
          "source": "src/Biobase-Secondary-Vienna.html#viennaPairTable",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "viennaPairTable",
          "package": "BiobaseXNA",
          "partial": "Pair Table",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:viennaPairTable"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary.Vienna",
          "name": "vpNP",
          "package": "BiobaseXNA",
          "signature": "ViennaPair",
          "source": "src/Biobase-Secondary-Vienna.html#vpNP",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary Vienna",
          "module": "Biobase.Secondary.Vienna",
          "name": "vpNP",
          "package": "BiobaseXNA",
          "partial": "NP",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary-Vienna.html#v:vpNP"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eSecondary structure: define basepairs as Int-tuples, the three edges, a\n nucleotide can use for pairing and the cis/trans isomerism. Both edges and\n cis/trans come with a tag for \u003ca\u003eunknown\u003c/a\u003e.\n\u003c/p\u003e\u003cp\u003eTODO set ext-annotations to be (isomerism,edge,edge) and have a asString\n instance to read \u003ca\u003ecWW\u003c/a\u003e \u003ca\u003etSH\u003c/a\u003e and other notation.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Biobase.Secondary",
          "name": "Secondary",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html",
          "type": "module"
        },
        "index": {
          "description": "Secondary structure define basepairs as Int-tuples the three edges nucleotide can use for pairing and the cis trans isomerism Both edges and cis trans come with tag for unknown TODO set ext-annotations to be isomerism edge edge and have asString instance to read cWW tSH and other notation",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "Secondary",
          "package": "BiobaseXNA",
          "partial": "Secondary",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eSelection of nucleotides and/or type classes independent of which type we\n are looking at.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "BaseSelect",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#BaseSelect",
          "type": "class"
        },
        "index": {
          "description": "Selection of nucleotides and or type classes independent of which type we are looking at",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "BaseSelect",
          "package": "BiobaseXNA",
          "partial": "Base Select",
          "type": "class"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:BaseSelect"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eNucleotides in a pairing may be in the cis(==?) or trans(==?) state.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "CTisomerism",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#CTisomerism",
          "type": "newtype"
        },
        "index": {
          "description": "Nucleotides in pairing may be in the cis or trans state",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "CTisomerism",
          "package": "BiobaseXNA",
          "partial": "CTisomerism",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:CTisomerism"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eEach nucleotide in a pair may be paired using one of three edges:\n watson-crik, sugar, or hoogsteen.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "Edge",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#Edge",
          "type": "newtype"
        },
        "index": {
          "description": "Each nucleotide in pair may be paired using one of three edges watson-crik sugar or hoogsteen",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "Edge",
          "package": "BiobaseXNA",
          "partial": "Edge",
          "type": "newtype"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:Edge"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eAn extended basepair, with nucleotides an annotation.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "ExtPair",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#ExtPair",
          "type": "type"
        },
        "index": {
          "description": "An extended basepair with nucleotides an annotation",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "ExtPair",
          "package": "BiobaseXNA",
          "partial": "Ext Pair",
          "type": "type"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:ExtPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eAnnotation for a basepair.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "ExtPairAnnotation",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#ExtPairAnnotation",
          "type": "type"
        },
        "index": {
          "description": "Annotation for basepair",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "ExtPairAnnotation",
          "package": "BiobaseXNA",
          "partial": "Ext Pair Annotation",
          "type": "type"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:ExtPairAnnotation"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eAn extended basepair is a basepair, annotated with edge and CTisomerism.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "ExtPairIdx",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#ExtPairIdx",
          "type": "type"
        },
        "index": {
          "description": "An extended basepair is basepair annotated with edge and CTisomerism",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "ExtPairIdx",
          "package": "BiobaseXNA",
          "partial": "Ext Pair Idx",
          "type": "type"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:ExtPairIdx"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eA pair as a tuple containing \u003ccode\u003e\u003ca\u003eNuc\u003c/a\u003e\u003c/code\u003es.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "Pair",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#Pair",
          "type": "type"
        },
        "index": {
          "description": "pair as tuple containing Nuc",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "Pair",
          "package": "BiobaseXNA",
          "partial": "Pair",
          "type": "type"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:Pair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eA basepair is simply a pair of Ints which are 0-indexing a sequence.\n\u003c/p\u003e\u003cp\u003eTODO storable vector, newtype, peek/poke?\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "PairIdx",
          "package": "BiobaseXNA",
          "source": "src/Biobase-Secondary.html#PairIdx",
          "type": "type"
        },
        "index": {
          "description": "basepair is simply pair of Ints which are indexing sequence TODO storable vector newtype peek poke",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "PairIdx",
          "package": "BiobaseXNA",
          "partial": "Pair Idx",
          "type": "type"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#t:PairIdx"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "CT",
          "package": "BiobaseXNA",
          "signature": "CT",
          "source": "src/Biobase-Secondary.html#CTisomerism",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "CT",
          "package": "BiobaseXNA",
          "partial": "CT",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:CT"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "Edge",
          "package": "BiobaseXNA",
          "signature": "Edge",
          "source": "src/Biobase-Secondary.html#Edge",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "Edge",
          "package": "BiobaseXNA",
          "partial": "Edge",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:Edge"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "antiCT",
          "package": "BiobaseXNA",
          "signature": "a",
          "source": "src/Biobase-Secondary.html#antiCT",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "antiCT",
          "package": "BiobaseXNA",
          "partial": "CT",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:antiCT"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eselect first index or nucleotide\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "baseL",
          "package": "BiobaseXNA",
          "signature": "a -\u003e b",
          "source": "src/Biobase-Secondary.html#baseL",
          "type": "method"
        },
        "index": {
          "description": "select first index or nucleotide",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "baseL",
          "normalized": "a-\u003eb",
          "package": "BiobaseXNA",
          "signature": "a-\u003eb",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:baseL"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eselect both nucleotides as pair\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "baseP",
          "package": "BiobaseXNA",
          "signature": "a -\u003e (b, b)",
          "source": "src/Biobase-Secondary.html#baseP",
          "type": "method"
        },
        "index": {
          "description": "select both nucleotides as pair",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "baseP",
          "normalized": "a-\u003e(b,b)",
          "package": "BiobaseXNA",
          "signature": "a-\u003e(b,b)",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:baseP"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eselect second index or nucleotide\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "baseR",
          "package": "BiobaseXNA",
          "signature": "a -\u003e b",
          "source": "src/Biobase-Secondary.html#baseR",
          "type": "method"
        },
        "index": {
          "description": "select second index or nucleotide",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "baseR",
          "normalized": "a-\u003eb",
          "package": "BiobaseXNA",
          "signature": "a-\u003eb",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:baseR"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eselect basepair type if existing or return default cWW\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "baseT",
          "package": "BiobaseXNA",
          "signature": "a -\u003e ExtPairAnnotation",
          "source": "src/Biobase-Secondary.html#baseT",
          "type": "method"
        },
        "index": {
          "description": "select basepair type if existing or return default cWW",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "baseT",
          "normalized": "a-\u003eExtPairAnnotation",
          "package": "BiobaseXNA",
          "signature": "a-\u003eExtPairAnnotation",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:baseT"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cHH",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cHH",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cHH",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "HH",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cHH"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cHS",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cHS",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cHS",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "HS",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cHS"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cHW",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cHW",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cHW",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "HW",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cHW"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cSH",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cSH",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cSH",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "SH",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cSH"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cSS",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cSS",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cSS",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "SS",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cSS"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cSW",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cSW",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cSW",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "SW",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cSW"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cWH",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cWH",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cWH",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "WH",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cWH"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cWS",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cWS",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cWS",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "WS",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cWS"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cWW",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#cWW",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cWW",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "WW",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cWW"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "charCTList",
          "package": "BiobaseXNA",
          "signature": "[(Char, CTisomerism)]",
          "source": "src/Biobase-Secondary.html#charCTList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "charCTList",
          "normalized": "[(Char,CTisomerism)]",
          "package": "BiobaseXNA",
          "partial": "CTList",
          "signature": "[(Char,CTisomerism)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:charCTList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "charEdgeList",
          "package": "BiobaseXNA",
          "signature": "[(Char, Edge)]",
          "source": "src/Biobase-Secondary.html#charEdgeList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "charEdgeList",
          "normalized": "[(Char,Edge)]",
          "package": "BiobaseXNA",
          "partial": "Edge List",
          "signature": "[(Char,Edge)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:charEdgeList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "cis",
          "package": "BiobaseXNA",
          "signature": "CTisomerism",
          "source": "src/Biobase-Secondary.html#cis",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "cis",
          "package": "BiobaseXNA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:cis"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "ctCharList",
          "package": "BiobaseXNA",
          "signature": "[(CTisomerism, Char)]",
          "source": "src/Biobase-Secondary.html#ctCharList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "ctCharList",
          "normalized": "[(CTisomerism,Char)]",
          "package": "BiobaseXNA",
          "partial": "Char List",
          "signature": "[(CTisomerism,Char)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:ctCharList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "edgeCharList",
          "package": "BiobaseXNA",
          "signature": "[(Edge, Char)]",
          "source": "src/Biobase-Secondary.html#edgeCharList",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "edgeCharList",
          "normalized": "[(Edge,Char)]",
          "package": "BiobaseXNA",
          "partial": "Char List",
          "signature": "[(Edge,Char)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:edgeCharList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "paraCT",
          "package": "BiobaseXNA",
          "signature": "a",
          "source": "src/Biobase-Secondary.html#paraCT",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "paraCT",
          "package": "BiobaseXNA",
          "partial": "CT",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:paraCT"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tHH",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tHH",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tHH",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "HH",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tHH"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tHS",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tHS",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tHS",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "HS",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tHS"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tHW",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tHW",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tHW",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "HW",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tHW"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tSH",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tSH",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tSH",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "SH",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tSH"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tSS",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tSS",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tSS",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "SS",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tSS"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tSW",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tSW",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tSW",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "SW",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tSW"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tWH",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tWH",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tWH",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "WH",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tWH"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tWS",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tWS",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tWS",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "WS",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tWS"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "tWW",
          "package": "BiobaseXNA",
          "signature": "(CTisomerism, Edge, Edge)",
          "source": "src/Biobase-Secondary.html#tWW",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "tWW",
          "normalized": "(CTisomerism,Edge,Edge)",
          "package": "BiobaseXNA",
          "partial": "WW",
          "signature": "(CTisomerism,Edge,Edge)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:tWW"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eEasy reading of a three-Char string into a triple.\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "threeChar",
          "package": "BiobaseXNA",
          "signature": "String -\u003e ExtPairAnnotation",
          "source": "src/Biobase-Secondary.html#threeChar",
          "type": "function"
        },
        "index": {
          "description": "Easy reading of three-Char string into triple",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "threeChar",
          "normalized": "String-\u003eExtPairAnnotation",
          "package": "BiobaseXNA",
          "partial": "Char",
          "signature": "String-\u003eExtPairAnnotation",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:threeChar"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "unCT",
          "package": "BiobaseXNA",
          "signature": "Int",
          "source": "src/Biobase-Secondary.html#CTisomerism",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "unCT",
          "package": "BiobaseXNA",
          "partial": "CT",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:unCT"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "unEdge",
          "package": "BiobaseXNA",
          "signature": "Int",
          "source": "src/Biobase-Secondary.html#Edge",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "unEdge",
          "package": "BiobaseXNA",
          "partial": "Edge",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:unEdge"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eupdate first index or nucleotide\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "updL",
          "package": "BiobaseXNA",
          "signature": "b -\u003e a -\u003e a",
          "source": "src/Biobase-Secondary.html#updL",
          "type": "method"
        },
        "index": {
          "description": "update first index or nucleotide",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "updL",
          "normalized": "a-\u003eb-\u003eb",
          "package": "BiobaseXNA",
          "signature": "b-\u003ea-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:updL"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eupdate complete pair\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "updP",
          "package": "BiobaseXNA",
          "signature": "(b, b) -\u003e a -\u003e a",
          "source": "src/Biobase-Secondary.html#updP",
          "type": "method"
        },
        "index": {
          "description": "update complete pair",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "updP",
          "normalized": "(a,a)-\u003eb-\u003eb",
          "package": "BiobaseXNA",
          "signature": "(b,b)-\u003ea-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:updP"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eupdate second index or nucleotide\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "updR",
          "package": "BiobaseXNA",
          "signature": "b -\u003e a -\u003e a",
          "source": "src/Biobase-Secondary.html#updR",
          "type": "method"
        },
        "index": {
          "description": "update second index or nucleotide",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "updR",
          "normalized": "a-\u003eb-\u003eb",
          "package": "BiobaseXNA",
          "signature": "b-\u003ea-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:updR"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eupdate basepair type, error if not possible due to type a\n\u003c/p\u003e",
          "module": "Biobase.Secondary",
          "name": "updT",
          "package": "BiobaseXNA",
          "signature": "ExtPairAnnotation -\u003e a -\u003e a",
          "source": "src/Biobase-Secondary.html#updT",
          "type": "method"
        },
        "index": {
          "description": "update basepair type error if not possible due to type",
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "updT",
          "normalized": "ExtPairAnnotation-\u003ea-\u003ea",
          "package": "BiobaseXNA",
          "signature": "ExtPairAnnotation-\u003ea-\u003ea",
          "type": "method"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:updT"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Biobase.Secondary",
          "name": "wc",
          "package": "BiobaseXNA",
          "signature": "Edge",
          "source": "src/Biobase-Secondary.html#wc",
          "type": "function"
        },
        "index": {
          "hierarchy": "Biobase Secondary",
          "module": "Biobase.Secondary",
          "name": "wc",
          "package": "BiobaseXNA",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/BiobaseXNA/docs/Biobase-Secondary.html#v:wc"
      }
    }
  ]
]