[
  {
    "cmd": "delete-by-query",
    "query": {
      "contexts": [
        "package"
      ],
      "query": {
        "op": "case",
        "type": "word",
        "word": "samtools"
      },
      "type": "context"
    }
  },
  [
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eThis module provides a fairly direct representation of the\n SAM/BAM alignment format, along with an interface to read and write\n alignments in this format.\n\u003c/p\u003e\u003cp\u003eThe package is based on the C SamTools library available at\n\u003c/p\u003e\u003cp\u003e\u003ca\u003ehttp://samtools.sourceforge.net/\u003c/a\u003e\n\u003c/p\u003e\u003cp\u003eand the SAM/BAM file format is described here\n\u003c/p\u003e\u003cp\u003e\u003ca\u003ehttp://samtools.sourceforge.net/SAM-1.3.pdf\u003c/a\u003e\n\u003c/p\u003e\u003cp\u003eThis package only reads existing alignment files generated by other\n tools. The meaning of the various flags is actually determined by\n the program that produced the alignment file.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "Bam",
          "package": "samtools",
          "source": "src/Bio-SamTools-Bam.html",
          "type": "module"
        },
        "index": {
          "description": "This module provides fairly direct representation of the SAM BAM alignment format along with an interface to read and write alignments in this format The package is based on the SamTools library available at http samtools.sourceforge.net and the SAM BAM file format is described here http samtools.sourceforge.net SAM-1.3.pdf This package only reads existing alignment files generated by other tools The meaning of the various flags is actually determined by the program that produced the alignment file",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "Bam",
          "package": "samtools",
          "partial": "Bam",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eSAM/BAM format alignment\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "Bam1",
          "package": "samtools",
          "source": "src/Bio-SamTools-Internal.html#Bam1",
          "type": "data"
        },
        "index": {
          "description": "SAM BAM format alignment",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "Bam1",
          "package": "samtools",
          "partial": "Bam",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#t:Bam1"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTarget sequences from a SAM alignment set\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "Header",
          "package": "samtools",
          "source": "src/Bio-SamTools-Internal.html#Header",
          "type": "data"
        },
        "index": {
          "description": "Target sequences from SAM alignment set",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "Header",
          "package": "samtools",
          "partial": "Header",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#t:Header"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eInformation about one target sequence in a SAM alignment set\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "HeaderSeq",
          "package": "samtools",
          "source": "src/Bio-SamTools-Internal.html#HeaderSeq",
          "type": "data"
        },
        "index": {
          "description": "Information about one target sequence in SAM alignment set",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "HeaderSeq",
          "package": "samtools",
          "partial": "Header Seq",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#t:HeaderSeq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eHandle for reading SAM/BAM format alignments\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "InHandle",
          "package": "samtools",
          "source": "src/Bio-SamTools-Bam.html#InHandle",
          "type": "data"
        },
        "index": {
          "description": "Handle for reading SAM BAM format alignments",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "InHandle",
          "package": "samtools",
          "partial": "In Handle",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#t:InHandle"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eHandle for writing SAM/BAM format alignments\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "OutHandle",
          "package": "samtools",
          "source": "src/Bio-SamTools-Bam.html#OutHandle",
          "type": "data"
        },
        "index": {
          "description": "Handle for writing SAM BAM format alignments",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "OutHandle",
          "package": "samtools",
          "partial": "Out Handle",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#t:OutHandle"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Bam",
          "name": "HeaderSeq",
          "package": "samtools",
          "signature": "HeaderSeq",
          "source": "src/Bio-SamTools-Internal.html#HeaderSeq",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "HeaderSeq",
          "package": "samtools",
          "partial": "Header Seq",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:HeaderSeq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eCIGAR description of the alignment\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "cigars",
          "package": "samtools",
          "signature": "Bam1 -\u003e [Cigar]",
          "source": "src/Bio-SamTools-Bam.html#cigars",
          "type": "function"
        },
        "index": {
          "description": "CIGAR description of the alignment",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "cigars",
          "normalized": "Bam-\u003e[Cigar]",
          "package": "samtools",
          "signature": "Bam-\u003e[Cigar]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:cigars"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eClose a SAM/BAM format alignment input handle\n\u003c/p\u003e\u003cp\u003eTarget sequence set data is still available after the file input\n has been closed.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "closeInHandle",
          "package": "samtools",
          "signature": "InHandle -\u003e IO ()",
          "source": "src/Bio-SamTools-Bam.html#closeInHandle",
          "type": "function"
        },
        "index": {
          "description": "Close SAM BAM format alignment input handle Target sequence set data is still available after the file input has been closed",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "closeInHandle",
          "normalized": "InHandle-\u003eIO()",
          "package": "samtools",
          "partial": "In Handle",
          "signature": "InHandle-\u003eIO()",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:closeInHandle"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eClose an alignment output handle\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "closeOutHandle",
          "package": "samtools",
          "signature": "OutHandle -\u003e IO ()",
          "source": "src/Bio-SamTools-Bam.html#closeOutHandle",
          "type": "function"
        },
        "index": {
          "description": "Close an alignment output handle",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "closeOutHandle",
          "normalized": "OutHandle-\u003eIO()",
          "package": "samtools",
          "partial": "Out Handle",
          "signature": "OutHandle-\u003eIO()",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:closeOutHandle"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eReads one alignment from an input handle, or returns \u003ccode\u003eNothing\u003c/code\u003e for end-of-file\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "get1",
          "package": "samtools",
          "signature": "InHandle -\u003e IO (Maybe Bam1)",
          "source": "src/Bio-SamTools-Bam.html#get1",
          "type": "function"
        },
        "index": {
          "description": "Reads one alignment from an input handle or returns Nothing for end-of-file",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "get1",
          "normalized": "InHandle-\u003eIO(Maybe Bam)",
          "package": "samtools",
          "signature": "InHandle-\u003eIO(Maybe Bam)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:get1"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTarget sequence set for the alignments\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "inHeader",
          "package": "samtools",
          "signature": "InHandle -\u003e Header",
          "source": "src/Bio-SamTools-Bam.html#inHeader",
          "type": "function"
        },
        "index": {
          "description": "Target sequence set for the alignments",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "inHeader",
          "normalized": "InHandle-\u003eHeader",
          "package": "samtools",
          "partial": "Header",
          "signature": "InHandle-\u003eHeader",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:inHeader"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the total insert length, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when the length is\n unavailable, e.g. because the read is unpaired or the mated read\n pair do not align in the proper relative orientation on the same\n strand.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "insertSize",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int64",
          "source": "src/Bio-SamTools-Bam.html#insertSize",
          "type": "function"
        },
        "index": {
          "description": "Just the total insert length or Nothing when the length is unavailable e.g because the read is unpaired or the mated read pair do not align in the proper relative orientation on the same strand",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "insertSize",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Size",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:insertSize"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the read a technical duplicate\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isDup",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isDup",
          "type": "function"
        },
        "index": {
          "description": "Is the read technical duplicate",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isDup",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Dup",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isDup"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the read paired and the mate's reverse complement aligned to the target\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isMateReverse",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isMateReverse",
          "type": "function"
        },
        "index": {
          "description": "Is the read paired and the mate reverse complement aligned to the target",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isMateReverse",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Mate Reverse",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isMateReverse"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the read paired and the mate unmapped\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isMateUnmap",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isMateUnmap",
          "type": "function"
        },
        "index": {
          "description": "Is the read paired and the mate unmapped",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isMateUnmap",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Mate Unmap",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isMateUnmap"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the read paired\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isPaired",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isPaired",
          "type": "function"
        },
        "index": {
          "description": "Is the read paired",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isPaired",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Paired",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isPaired"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the pair properly aligned (usually based on relative orientation and distance)\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isProperPair",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isProperPair",
          "type": "function"
        },
        "index": {
          "description": "Is the pair properly aligned usually based on relative orientation and distance",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isProperPair",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Proper Pair",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isProperPair"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eDid the read fail quality controls\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isQCFail",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isQCFail",
          "type": "function"
        },
        "index": {
          "description": "Did the read fail quality controls",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isQCFail",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "QCFail",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isQCFail"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the fragment from the first read in the template\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isRead1",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isRead1",
          "type": "function"
        },
        "index": {
          "description": "Is the fragment from the first read in the template",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isRead1",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Read",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isRead1"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the fragment from the second read in the template\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isRead2",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isRead2",
          "type": "function"
        },
        "index": {
          "description": "Is the fragment from the second read in the template",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isRead2",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Read",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isRead2"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the fragment's reverse complement aligned to the target\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isReverse",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isReverse",
          "type": "function"
        },
        "index": {
          "description": "Is the fragment reverse complement aligned to the target",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isReverse",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Reverse",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isReverse"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the fragment alignment secondary\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isSecondary",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isSecondary",
          "type": "function"
        },
        "index": {
          "description": "Is the fragment alignment secondary",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isSecondary",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Secondary",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isSecondary"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eIs the read unmapped\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "isUnmap",
          "package": "samtools",
          "signature": "Bam1 -\u003e Bool",
          "source": "src/Bio-SamTools-Bam.html#isUnmap",
          "type": "function"
        },
        "index": {
          "description": "Is the read unmapped",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "isUnmap",
          "normalized": "Bam-\u003eBool",
          "package": "samtools",
          "partial": "Unmap",
          "signature": "Bam-\u003eBool",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:isUnmap"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTarget sequence lengh\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "len",
          "package": "samtools",
          "signature": "Int64",
          "source": "src/Bio-SamTools-Internal.html#HeaderSeq",
          "type": "function"
        },
        "index": {
          "description": "Target sequence lengh",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "len",
          "package": "samtools",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:len"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Bam",
          "name": "lookupTarget",
          "package": "samtools",
          "signature": "Header -\u003e ByteString -\u003e Maybe Int",
          "source": "src/Bio-SamTools-Internal.html#lookupTarget",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "lookupTarget",
          "normalized": "Header-\u003eByteString-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Target",
          "signature": "Header-\u003eByteString-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:lookupTarget"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the match descriptor alignment field, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when it\n is absent\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "matchDesc",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe ByteString",
          "source": "src/Bio-SamTools-Bam.html#matchDesc",
          "type": "function"
        },
        "index": {
          "description": "Just the match descriptor alignment field or Nothing when it is absent",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "matchDesc",
          "normalized": "Bam-\u003eMaybe ByteString",
          "package": "samtools",
          "partial": "Desc",
          "signature": "Bam-\u003eMaybe ByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:matchDesc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e'Just the 0-based coordinate of the left-most position in the\n mate alignment on the target, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when the read is\n unpaired or the mate is unmapped.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "matePosition",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int64",
          "source": "src/Bio-SamTools-Bam.html#matePosition",
          "type": "function"
        },
        "index": {
          "description": "Just the based coordinate of the left-most position in the mate alignment on the target or Nothing when the read is unpaired or the mate is unmapped",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "matePosition",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Position",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:matePosition"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the target ID of the mate alignment target reference\n sequence, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when the mate is unmapped or the read is\n unpaired.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "mateTargetID",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int",
          "source": "src/Bio-SamTools-Bam.html#mateTargetID",
          "type": "function"
        },
        "index": {
          "description": "Just the target ID of the mate alignment target reference sequence or Nothing when the mate is unmapped or the read is unpaired",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "mateTargetID",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Target ID",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:mateTargetID"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the length of the mate alignment target reference\n sequence, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when the mate is unmapped or the read is\n unpaired.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "mateTargetLen",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int64",
          "source": "src/Bio-SamTools-Bam.html#mateTargetLen",
          "type": "function"
        },
        "index": {
          "description": "Just the length of the mate alignment target reference sequence or Nothing when the mate is unmapped or the read is unpaired",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "mateTargetLen",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Target Len",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:mateTargetLen"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the name of the mate alignment target reference sequence,\n or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when the mate is unmapped or the read is unpaired.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "mateTargetName",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe ByteString",
          "source": "src/Bio-SamTools-Bam.html#mateTargetName",
          "type": "function"
        },
        "index": {
          "description": "Just the name of the mate alignment target reference sequence or Nothing when the mate is unmapped or the read is unpaired",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "mateTargetName",
          "normalized": "Bam-\u003eMaybe ByteString",
          "package": "samtools",
          "partial": "Target Name",
          "signature": "Bam-\u003eMaybe ByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:mateTargetName"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the number of reported alignments, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when this\n information is not present.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "nHits",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int",
          "source": "src/Bio-SamTools-Bam.html#nHits",
          "type": "function"
        },
        "index": {
          "description": "Just the number of reported alignments or Nothing when this information is not present",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "nHits",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Hits",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:nHits"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the number of mismatches in the alignemnt, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e\n when this information is not present\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "nMismatch",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int",
          "source": "src/Bio-SamTools-Bam.html#nMismatch",
          "type": "function"
        },
        "index": {
          "description": "Just the number of mismatches in the alignemnt or Nothing when this information is not present",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "nMismatch",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Mismatch",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:nMismatch"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eNumber of target sequences\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "nTargets",
          "package": "samtools",
          "signature": "Header -\u003e Int",
          "source": "src/Bio-SamTools-Internal.html#nTargets",
          "type": "function"
        },
        "index": {
          "description": "Number of target sequences",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "nTargets",
          "normalized": "Header-\u003eInt",
          "package": "samtools",
          "partial": "Targets",
          "signature": "Header-\u003eInt",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:nTargets"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTarget sequence name \n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "name",
          "package": "samtools",
          "signature": "ByteString",
          "source": "src/Bio-SamTools-Internal.html#HeaderSeq",
          "type": "function"
        },
        "index": {
          "description": "Target sequence name",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "name",
          "package": "samtools",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:name"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eOpen a BAM (binary) format file\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "openBamInFile",
          "package": "samtools",
          "signature": "FilePath -\u003e IO InHandle",
          "source": "src/Bio-SamTools-Bam.html#openBamInFile",
          "type": "function"
        },
        "index": {
          "description": "Open BAM binary format file",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "openBamInFile",
          "normalized": "FilePath-\u003eIO InHandle",
          "package": "samtools",
          "partial": "Bam In File",
          "signature": "FilePath-\u003eIO InHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:openBamInFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eOpen a BAM format file for writing alignments\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "openBamOutFile",
          "package": "samtools",
          "signature": "FilePath -\u003e Header -\u003e IO OutHandle",
          "source": "src/Bio-SamTools-Bam.html#openBamOutFile",
          "type": "function"
        },
        "index": {
          "description": "Open BAM format file for writing alignments",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "openBamOutFile",
          "normalized": "FilePath-\u003eHeader-\u003eIO OutHandle",
          "package": "samtools",
          "partial": "Bam Out File",
          "signature": "FilePath-\u003eHeader-\u003eIO OutHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:openBamOutFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eOpen a TAM (tab-delimited text) format file with \u003ccode\u003e@SQ\u003c/code\u003e headers\n for the target sequence set.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "openTamInFile",
          "package": "samtools",
          "signature": "FilePath -\u003e IO InHandle",
          "source": "src/Bio-SamTools-Bam.html#openTamInFile",
          "type": "function"
        },
        "index": {
          "description": "Open TAM tab-delimited text format file with SQ headers for the target sequence set",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "openTamInFile",
          "normalized": "FilePath-\u003eIO InHandle",
          "package": "samtools",
          "partial": "Tam In File",
          "signature": "FilePath-\u003eIO InHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:openTamInFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eOpen a TAM format file with a separate target sequence set index\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "openTamInFileWithIndex",
          "package": "samtools",
          "signature": "FilePath -\u003e FilePath -\u003e IO InHandle",
          "source": "src/Bio-SamTools-Bam.html#openTamInFileWithIndex",
          "type": "function"
        },
        "index": {
          "description": "Open TAM format file with separate target sequence set index",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "openTamInFileWithIndex",
          "normalized": "FilePath-\u003eFilePath-\u003eIO InHandle",
          "package": "samtools",
          "partial": "Tam In File With Index",
          "signature": "FilePath-\u003eFilePath-\u003eIO InHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:openTamInFileWithIndex"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eOpen a TAM format file with \u003ccode\u003e@SQ\u003c/code\u003e headers for writing alignments\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "openTamOutFile",
          "package": "samtools",
          "signature": "FilePath -\u003e Header -\u003e IO OutHandle",
          "source": "src/Bio-SamTools-Bam.html#openTamOutFile",
          "type": "function"
        },
        "index": {
          "description": "Open TAM format file with SQ headers for writing alignments",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "openTamOutFile",
          "normalized": "FilePath-\u003eHeader-\u003eIO OutHandle",
          "package": "samtools",
          "partial": "Tam Out File",
          "signature": "FilePath-\u003eHeader-\u003eIO OutHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:openTamOutFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTarget sequence set for the alignments\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "outHeader",
          "package": "samtools",
          "signature": "OutHandle -\u003e Header",
          "source": "src/Bio-SamTools-Bam.html#outHeader",
          "type": "function"
        },
        "index": {
          "description": "Target sequence set for the alignments",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "outHeader",
          "normalized": "OutHandle-\u003eHeader",
          "package": "samtools",
          "partial": "Header",
          "signature": "OutHandle-\u003eHeader",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:outHeader"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the 0-based index of the leftmost aligned position on the\n target sequence, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e for an unmapped read\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "position",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int64",
          "source": "src/Bio-SamTools-Bam.html#position",
          "type": "function"
        },
        "index": {
          "description": "Just the based index of the leftmost aligned position on the target sequence or Nothing for an unmapped read",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "position",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:position"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eWrites one alignment to an input handle.\n\u003c/p\u003e\u003cp\u003eThere is no validation that the target sequence set of the output\n handle matches the target sequence set of the alignment.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "put1",
          "package": "samtools",
          "signature": "OutHandle -\u003e Bam1 -\u003e IO ()",
          "source": "src/Bio-SamTools-Bam.html#put1",
          "type": "function"
        },
        "index": {
          "description": "Writes one alignment to an input handle There is no validation that the target sequence set of the output handle matches the target sequence set of the alignment",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "put1",
          "normalized": "OutHandle-\u003eBam-\u003eIO()",
          "package": "samtools",
          "signature": "OutHandle-\u003eBam-\u003eIO()",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:put1"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the length of the query sequence, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when it is\n unavailable.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "queryLength",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int64",
          "source": "src/Bio-SamTools-Bam.html#queryLength",
          "type": "function"
        },
        "index": {
          "description": "Just the length of the query sequence or Nothing when it is unavailable",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "queryLength",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Length",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:queryLength"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eName of the query sequence\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "queryName",
          "package": "samtools",
          "signature": "Bam1 -\u003e ByteString",
          "source": "src/Bio-SamTools-Bam.html#queryName",
          "type": "function"
        },
        "index": {
          "description": "Name of the query sequence",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "queryName",
          "normalized": "Bam-\u003eByteString",
          "package": "samtools",
          "partial": "Name",
          "signature": "Bam-\u003eByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:queryName"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the query qualities, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when it is\n unavailable. These are returned in ASCII format, i.e., \u003cem\u003eq\u003c/em\u003e + 33.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "queryQual",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe ByteString",
          "source": "src/Bio-SamTools-Bam.html#queryQual",
          "type": "function"
        },
        "index": {
          "description": "Just the query qualities or Nothing when it is unavailable These are returned in ASCII format i.e",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "queryQual",
          "normalized": "Bam-\u003eMaybe ByteString",
          "package": "samtools",
          "partial": "Qual",
          "signature": "Bam-\u003eMaybe ByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:queryQual"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the query sequence, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e when it is unavailable\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "querySeq",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe ByteString",
          "source": "src/Bio-SamTools-Bam.html#querySeq",
          "type": "function"
        },
        "index": {
          "description": "Just the query sequence or Nothing when it is unavailable",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "querySeq",
          "normalized": "Bam-\u003eMaybe ByteString",
          "package": "samtools",
          "partial": "Seq",
          "signature": "Bam-\u003eMaybe ByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:querySeq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eRead a BAM file as a lazy strem of \u003ccode\u003e\u003ca\u003eBam1\u003c/a\u003e\u003c/code\u003e records.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "readBams",
          "package": "samtools",
          "signature": "FilePath -\u003e IO [Bam1]",
          "source": "src/Bio-SamTools-Bam.html#readBams",
          "type": "function"
        },
        "index": {
          "description": "Read BAM file as lazy strem of Bam1 records",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "readBams",
          "normalized": "FilePath-\u003eIO[Bam]",
          "package": "samtools",
          "partial": "Bams",
          "signature": "FilePath-\u003eIO[Bam]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:readBams"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the reference sequence location (as per \u003ccode\u003e\u003ca\u003erefSpLoc\u003c/a\u003e\u003c/code\u003e) on\n the target reference (as per \u003ccode\u003e\u003ca\u003etargetName\u003c/a\u003e\u003c/code\u003e)\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "refSeqLoc",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe SpliceSeqLoc",
          "source": "src/Bio-SamTools-Bam.html#refSeqLoc",
          "type": "function"
        },
        "index": {
          "description": "Just the reference sequence location as per refSpLoc on the target reference as per targetName",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "refSeqLoc",
          "normalized": "Bam-\u003eMaybe SpliceSeqLoc",
          "package": "samtools",
          "partial": "Seq Loc",
          "signature": "Bam-\u003eMaybe SpliceSeqLoc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:refSeqLoc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the reference sequence location covered by the\n alignment. This includes nucleotide positions that are reported to\n be deleted in the read, but not skipped nucleotide position\n (typically intronic positions in a spliced alignment). If the\n reference location is unavailable, e.g. for an unmapped read or for\n a read with no CIGAR format alignment information, then \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "refSpLoc",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe SpliceLoc",
          "source": "src/Bio-SamTools-Bam.html#refSpLoc",
          "type": "function"
        },
        "index": {
          "description": "Just the reference sequence location covered by the alignment This includes nucleotide positions that are reported to be deleted in the read but not skipped nucleotide position typically intronic positions in spliced alignment If the reference location is unavailable e.g for an unmapped read or for read with no CIGAR format alignment information then Nothing",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "refSpLoc",
          "normalized": "Bam-\u003eMaybe SpliceLoc",
          "package": "samtools",
          "partial": "Sp Loc",
          "signature": "Bam-\u003eMaybe SpliceLoc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:refSpLoc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the reference target sequence ID in the target set, or\n \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e for an unmapped read\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "targetID",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int",
          "source": "src/Bio-SamTools-Bam.html#targetID",
          "type": "function"
        },
        "index": {
          "description": "Just the reference target sequence ID in the target set or Nothing for an unmapped read",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "targetID",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "ID",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:targetID"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the total length of the target sequence, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e for\n an unmapped read\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "targetLen",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe Int64",
          "source": "src/Bio-SamTools-Bam.html#targetLen",
          "type": "function"
        },
        "index": {
          "description": "Just the total length of the target sequence or Nothing for an unmapped read",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "targetLen",
          "normalized": "Bam-\u003eMaybe Int",
          "package": "samtools",
          "partial": "Len",
          "signature": "Bam-\u003eMaybe Int",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:targetLen"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003e\u003ccode\u003e\u003ca\u003eJust\u003c/a\u003e\u003c/code\u003e the target sequence name, or \u003ccode\u003e\u003ca\u003eNothing\u003c/a\u003e\u003c/code\u003e for an unmapped\n read\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "targetName",
          "package": "samtools",
          "signature": "Bam1 -\u003e Maybe ByteString",
          "source": "src/Bio-SamTools-Bam.html#targetName",
          "type": "function"
        },
        "index": {
          "description": "Just the target sequence name or Nothing for an unmapped read",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "targetName",
          "normalized": "Bam-\u003eMaybe ByteString",
          "package": "samtools",
          "partial": "Name",
          "signature": "Bam-\u003eMaybe ByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:targetName"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eReturns a target sequence by ID, which is a 0-based index\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "targetSeq",
          "package": "samtools",
          "signature": "Header -\u003e Int -\u003e HeaderSeq",
          "source": "src/Bio-SamTools-Internal.html#targetSeq",
          "type": "function"
        },
        "index": {
          "description": "Returns target sequence by ID which is based index",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "targetSeq",
          "normalized": "Header-\u003eInt-\u003eHeaderSeq",
          "package": "samtools",
          "partial": "Seq",
          "signature": "Header-\u003eInt-\u003eHeaderSeq",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:targetSeq"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Bam",
          "name": "targetSeqLen",
          "package": "samtools",
          "signature": "Header -\u003e Int -\u003e Int64",
          "source": "src/Bio-SamTools-Internal.html#targetSeqLen",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "targetSeqLen",
          "normalized": "Header-\u003eInt-\u003eInt",
          "package": "samtools",
          "partial": "Seq Len",
          "signature": "Header-\u003eInt-\u003eInt",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:targetSeqLen"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eReturns the list of target sequences\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "targetSeqList",
          "package": "samtools",
          "signature": "Header -\u003e [HeaderSeq]",
          "source": "src/Bio-SamTools-Internal.html#targetSeqList",
          "type": "function"
        },
        "index": {
          "description": "Returns the list of target sequences",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "targetSeqList",
          "normalized": "Header-\u003e[HeaderSeq]",
          "package": "samtools",
          "partial": "Seq List",
          "signature": "Header-\u003e[HeaderSeq]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:targetSeqList"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eReturns a target sequence name by ID\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "targetSeqName",
          "package": "samtools",
          "signature": "Header -\u003e Int -\u003e ByteString",
          "source": "src/Bio-SamTools-Internal.html#targetSeqName",
          "type": "function"
        },
        "index": {
          "description": "Returns target sequence name by ID",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "targetSeqName",
          "normalized": "Header-\u003eInt-\u003eByteString",
          "package": "samtools",
          "partial": "Seq Name",
          "signature": "Header-\u003eInt-\u003eByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:targetSeqName"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eAs \u003ccode\u003e\u003ca\u003ewithTamInFile\u003c/a\u003e\u003c/code\u003e for BAM (binary) format files\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "withBamInFile",
          "package": "samtools",
          "signature": "FilePath -\u003e (InHandle -\u003e IO a) -\u003e IO a",
          "source": "src/Bio-SamTools-Bam.html#withBamInFile",
          "type": "function"
        },
        "index": {
          "description": "As withTamInFile for BAM binary format files",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "withBamInFile",
          "normalized": "FilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "package": "samtools",
          "partial": "Bam In File",
          "signature": "FilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:withBamInFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Bam",
          "name": "withBamOutFile",
          "package": "samtools",
          "signature": "FilePath -\u003e Header -\u003e (OutHandle -\u003e IO a) -\u003e IO a",
          "source": "src/Bio-SamTools-Bam.html#withBamOutFile",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "withBamOutFile",
          "normalized": "FilePath-\u003eHeader-\u003e(OutHandle-\u003eIO a)-\u003eIO a",
          "package": "samtools",
          "partial": "Bam Out File",
          "signature": "FilePath-\u003eHeader-\u003e(OutHandle-\u003eIO a)-\u003eIO a",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:withBamOutFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eRun an IO action using a handle to a TAM format file that will be\n opened (see \u003ccode\u003e\u003ca\u003eopenTamInFile\u003c/a\u003e\u003c/code\u003e) and closed for the action.\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "withTamInFile",
          "package": "samtools",
          "signature": "FilePath -\u003e (InHandle -\u003e IO a) -\u003e IO a",
          "source": "src/Bio-SamTools-Bam.html#withTamInFile",
          "type": "function"
        },
        "index": {
          "description": "Run an IO action using handle to TAM format file that will be opened see openTamInFile and closed for the action",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "withTamInFile",
          "normalized": "FilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "package": "samtools",
          "partial": "Tam In File",
          "signature": "FilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:withTamInFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eAs \u003ccode\u003e\u003ca\u003ewithTamInFile\u003c/a\u003e\u003c/code\u003e with a separate target sequence index set (see\n \u003ccode\u003e\u003ca\u003eopenTamInFileWithIndex\u003c/a\u003e\u003c/code\u003e)\n\u003c/p\u003e",
          "module": "Bio.SamTools.Bam",
          "name": "withTamInFileWithIndex",
          "package": "samtools",
          "signature": "FilePath -\u003e FilePath -\u003e (InHandle -\u003e IO a) -\u003e IO a",
          "source": "src/Bio-SamTools-Bam.html#withTamInFileWithIndex",
          "type": "function"
        },
        "index": {
          "description": "As withTamInFile with separate target sequence index set see openTamInFileWithIndex",
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "withTamInFileWithIndex",
          "normalized": "FilePath-\u003eFilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "package": "samtools",
          "partial": "Tam In File With Index",
          "signature": "FilePath-\u003eFilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:withTamInFileWithIndex"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Bam",
          "name": "withTamOutFile",
          "package": "samtools",
          "signature": "FilePath -\u003e Header -\u003e (OutHandle -\u003e IO a) -\u003e IO a",
          "source": "src/Bio-SamTools-Bam.html#withTamOutFile",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Bam",
          "module": "Bio.SamTools.Bam",
          "name": "withTamOutFile",
          "normalized": "FilePath-\u003eHeader-\u003e(OutHandle-\u003eIO a)-\u003eIO a",
          "package": "samtools",
          "partial": "Tam Out File",
          "signature": "FilePath-\u003eHeader-\u003e(OutHandle-\u003eIO a)-\u003eIO a",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Bam.html#v:withTamOutFile"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eThis module provides an interface to sorted, indexed BAM\n alignment files, which allow rapid extraction of alignments that lie\n within one specific region of one sequence.\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Bio.SamTools.BamIndex",
          "name": "BamIndex",
          "package": "samtools",
          "source": "src/Bio-SamTools-BamIndex.html",
          "type": "module"
        },
        "index": {
          "description": "This module provides an interface to sorted indexed BAM alignment files which allow rapid extraction of alignments that lie within one specific region of one sequence",
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "BamIndex",
          "package": "samtools",
          "partial": "Bam Index",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eHandle for fetching alignments by region from a sorted, indexed\n BAM file.\n\u003c/p\u003e",
          "module": "Bio.SamTools.BamIndex",
          "name": "IdxHandle",
          "package": "samtools",
          "source": "src/Bio-SamTools-BamIndex.html#IdxHandle",
          "type": "data"
        },
        "index": {
          "description": "Handle for fetching alignments by region from sorted indexed BAM file",
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "IdxHandle",
          "package": "samtools",
          "partial": "Idx Handle",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#t:IdxHandle"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.BamIndex",
          "name": "Query",
          "package": "samtools",
          "source": "src/Bio-SamTools-BamIndex.html#Query",
          "type": "data"
        },
        "index": {
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "Query",
          "package": "samtools",
          "partial": "Query",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#t:Query"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.BamIndex",
          "name": "close",
          "package": "samtools",
          "signature": "IdxHandle -\u003e IO ()",
          "source": "src/Bio-SamTools-BamIndex.html#close",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "close",
          "normalized": "IdxHandle-\u003eIO()",
          "package": "samtools",
          "signature": "IdxHandle-\u003eIO()",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:close"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eFilename of sorted, indexed BAM file\n\u003c/p\u003e",
          "module": "Bio.SamTools.BamIndex",
          "name": "idxFilename",
          "package": "samtools",
          "signature": "IdxHandle -\u003e FilePath",
          "source": "src/Bio-SamTools-BamIndex.html#idxFilename",
          "type": "function"
        },
        "index": {
          "description": "Filename of sorted indexed BAM file",
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "idxFilename",
          "normalized": "IdxHandle-\u003eFilePath",
          "package": "samtools",
          "partial": "Filename",
          "signature": "IdxHandle-\u003eFilePath",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:idxFilename"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTarget sequences\n\u003c/p\u003e",
          "module": "Bio.SamTools.BamIndex",
          "name": "idxHeader",
          "package": "samtools",
          "signature": "IdxHandle -\u003e Header",
          "source": "src/Bio-SamTools-BamIndex.html#idxHeader",
          "type": "function"
        },
        "index": {
          "description": "Target sequences",
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "idxHeader",
          "normalized": "IdxHandle-\u003eHeader",
          "package": "samtools",
          "partial": "Header",
          "signature": "IdxHandle-\u003eHeader",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:idxHeader"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.BamIndex",
          "name": "next",
          "package": "samtools",
          "signature": "Query -\u003e IO (Maybe Bam1)",
          "source": "src/Bio-SamTools-BamIndex.html#next",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "next",
          "normalized": "Query-\u003eIO(Maybe Bam)",
          "package": "samtools",
          "signature": "Query-\u003eIO(Maybe Bam)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:next"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eOpen a sorted, indexed BAM file.              \n\u003c/p\u003e",
          "module": "Bio.SamTools.BamIndex",
          "name": "open",
          "package": "samtools",
          "signature": "FilePath -\u003e IO IdxHandle",
          "source": "src/Bio-SamTools-BamIndex.html#open",
          "type": "function"
        },
        "index": {
          "description": "Open sorted indexed BAM file",
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "open",
          "normalized": "FilePath-\u003eIO IdxHandle",
          "package": "samtools",
          "signature": "FilePath-\u003eIO IdxHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:open"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.BamIndex",
          "name": "query",
          "package": "samtools",
          "signature": "IdxHandle -\u003e Int -\u003e (Int64, Int64) -\u003e IO Query",
          "source": "src/Bio-SamTools-BamIndex.html#query",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "query",
          "normalized": "IdxHandle-\u003eInt-\u003e(Int,Int)-\u003eIO Query",
          "package": "samtools",
          "signature": "IdxHandle-\u003eInt-\u003e(Int,Int)-\u003eIO Query",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:query"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.BamIndex",
          "name": "qyHandle",
          "package": "samtools",
          "signature": "Query -\u003e IdxHandle",
          "source": "src/Bio-SamTools-BamIndex.html#qyHandle",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "qyHandle",
          "normalized": "Query-\u003eIdxHandle",
          "package": "samtools",
          "partial": "Handle",
          "signature": "Query-\u003eIdxHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:qyHandle"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eUse a BAM index file to extract \u003ccode\u003e\u003ca\u003eBam1\u003c/a\u003e\u003c/code\u003e records aligned to a \n specific target sequence (chromosome) number and region.\n\u003c/p\u003e",
          "module": "Bio.SamTools.BamIndex",
          "name": "readBamRegion",
          "package": "samtools",
          "signature": "IdxHandle -\u003e Int -\u003e (Int64, Int64) -\u003e IO [Bam1]",
          "source": "src/Bio-SamTools-BamIndex.html#readBamRegion",
          "type": "function"
        },
        "index": {
          "description": "Use BAM index file to extract Bam1 records aligned to specific target sequence chromosome number and region",
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "readBamRegion",
          "normalized": "IdxHandle-\u003eInt-\u003e(Int,Int)-\u003eIO[Bam]",
          "package": "samtools",
          "partial": "Bam Region",
          "signature": "IdxHandle-\u003eInt-\u003e(Int,Int)-\u003eIO[Bam]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:readBamRegion"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.BamIndex",
          "name": "withIndex",
          "package": "samtools",
          "signature": "FilePath -\u003e (IdxHandle -\u003e IO a) -\u003e IO a",
          "source": "src/Bio-SamTools-BamIndex.html#withIndex",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools BamIndex",
          "module": "Bio.SamTools.BamIndex",
          "name": "withIndex",
          "normalized": "FilePath-\u003e(IdxHandle-\u003eIO a)-\u003eIO a",
          "package": "samtools",
          "partial": "Index",
          "signature": "FilePath-\u003e(IdxHandle-\u003eIO a)-\u003eIO a",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-BamIndex.html#v:withIndex"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eHandling of the extended CIGAR pair-wise alignment descriptors\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "Cigar",
          "package": "samtools",
          "source": "src/Bio-SamTools-Cigar.html",
          "type": "module"
        },
        "index": {
          "description": "Handling of the extended CIGAR pair-wise alignment descriptors",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "Cigar",
          "package": "samtools",
          "partial": "Cigar",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eCigar entry including length\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "Cigar",
          "package": "samtools",
          "source": "src/Bio-SamTools-Cigar.html#Cigar",
          "type": "data"
        },
        "index": {
          "description": "Cigar entry including length",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "Cigar",
          "package": "samtools",
          "partial": "Cigar",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#t:Cigar"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eCigar entry types\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "CigarType",
          "package": "samtools",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "data"
        },
        "index": {
          "description": "Cigar entry types",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "CigarType",
          "package": "samtools",
          "partial": "Cigar Type",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#t:CigarType"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Cigar",
          "name": "Cigar",
          "package": "samtools",
          "signature": "Cigar",
          "source": "src/Bio-SamTools-Cigar.html#Cigar",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "Cigar",
          "package": "samtools",
          "partial": "Cigar",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:Cigar"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eDeletion from read relative to reference\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "Del",
          "package": "samtools",
          "signature": "Del",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "function"
        },
        "index": {
          "description": "Deletion from read relative to reference",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "Del",
          "package": "samtools",
          "partial": "Del",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:Del"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTrimmed nucleotides, removed from read\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "HardClip",
          "package": "samtools",
          "signature": "HardClip",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "function"
        },
        "index": {
          "description": "Trimmed nucleotides removed from read",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "HardClip",
          "package": "samtools",
          "partial": "Hard Clip",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:HardClip"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eInsertion in read relative to reference\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "Ins",
          "package": "samtools",
          "signature": "Ins",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "function"
        },
        "index": {
          "description": "Insertion in read relative to reference",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "Ins",
          "package": "samtools",
          "partial": "Ins",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:Ins"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eAligned nucleotide, may be a match or mismatch\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "Match",
          "package": "samtools",
          "signature": "Match",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "function"
        },
        "index": {
          "description": "Aligned nucleotide may be match or mismatch",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "Match",
          "package": "samtools",
          "partial": "Match",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:Match"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eDeletion from padded reference\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "Pad",
          "package": "samtools",
          "signature": "Pad",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "function"
        },
        "index": {
          "description": "Deletion from padded reference",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "Pad",
          "package": "samtools",
          "partial": "Pad",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:Pad"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eSkipped reference bases, i.e., splice\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "RefSkip",
          "package": "samtools",
          "signature": "RefSkip",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "function"
        },
        "index": {
          "description": "Skipped reference bases i.e splice",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "RefSkip",
          "package": "samtools",
          "partial": "Ref Skip",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:RefSkip"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eTrimmed nucleotides, still present in read\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "SoftClip",
          "package": "samtools",
          "signature": "SoftClip",
          "source": "src/Bio-SamTools-Cigar.html#CigarType",
          "type": "function"
        },
        "index": {
          "description": "Trimmed nucleotides still present in read",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "SoftClip",
          "package": "samtools",
          "partial": "Soft Clip",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:SoftClip"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Cigar",
          "name": "cigar",
          "package": "samtools",
          "signature": "CigarType",
          "source": "src/Bio-SamTools-Cigar.html#Cigar",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "cigar",
          "package": "samtools",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:cigar"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Cigar",
          "name": "cigarToAlignment",
          "package": "samtools",
          "signature": "Int64 -\u003e [Cigar] -\u003e [(Maybe Pos, Maybe Pos)]",
          "source": "src/Bio-SamTools-Cigar.html#cigarToAlignment",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "cigarToAlignment",
          "normalized": "Int-\u003e[Cigar]-\u003e[(Maybe Pos,Maybe Pos)]",
          "package": "samtools",
          "partial": "To Alignment",
          "signature": "Int-\u003e[Cigar]-\u003e[(Maybe Pos,Maybe Pos)]",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:cigarToAlignment"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Cigar",
          "name": "cigarToSpLoc",
          "package": "samtools",
          "signature": "Int64 -\u003e [Cigar] -\u003e SpliceLoc",
          "source": "src/Bio-SamTools-Cigar.html#cigarToSpLoc",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "cigarToSpLoc",
          "normalized": "Int-\u003e[Cigar]-\u003eSpliceLoc",
          "package": "samtools",
          "partial": "To Sp Loc",
          "signature": "Int-\u003e[Cigar]-\u003eSpliceLoc",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:cigarToSpLoc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.Cigar",
          "name": "length",
          "package": "samtools",
          "signature": "Int64",
          "source": "src/Bio-SamTools-Cigar.html#Cigar",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "length",
          "package": "samtools",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:length"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eConvert a \u003ccode\u003eBAM\u003c/code\u003e binary cigar integer to a \u003ccode\u003e\u003ca\u003eCigar\u003c/a\u003e\u003c/code\u003e\n\u003c/p\u003e",
          "module": "Bio.SamTools.Cigar",
          "name": "toCigar",
          "package": "samtools",
          "signature": "CUInt -\u003e Cigar",
          "source": "src/Bio-SamTools-Cigar.html#toCigar",
          "type": "function"
        },
        "index": {
          "description": "Convert BAM binary cigar integer to Cigar",
          "hierarchy": "Bio SamTools Cigar",
          "module": "Bio.SamTools.Cigar",
          "name": "toCigar",
          "normalized": "CUInt-\u003eCigar",
          "package": "samtools",
          "partial": "Cigar",
          "signature": "CUInt-\u003eCigar",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-Cigar.html#v:toCigar"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cdiv class=\"doc\"\u003e\u003cp\u003eFetch sequences from an indexed fasta file\n\u003c/p\u003e\u003c/div\u003e",
          "module": "Bio.SamTools.FaIdx",
          "name": "FaIdx",
          "package": "samtools",
          "source": "src/Bio-SamTools-FaIdx.html",
          "type": "module"
        },
        "index": {
          "description": "Fetch sequences from an indexed fasta file",
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "FaIdx",
          "package": "samtools",
          "partial": "Fa Idx",
          "type": "module"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eInput handle for an indexed fasta file\n\u003c/p\u003e",
          "module": "Bio.SamTools.FaIdx",
          "name": "InHandle",
          "package": "samtools",
          "source": "src/Bio-SamTools-FaIdx.html#InHandle",
          "type": "data"
        },
        "index": {
          "description": "Input handle for an indexed fasta file",
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "InHandle",
          "package": "samtools",
          "partial": "In Handle",
          "type": "data"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#t:InHandle"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.FaIdx",
          "name": "close",
          "package": "samtools",
          "signature": "InHandle -\u003e IO ()",
          "source": "src/Bio-SamTools-FaIdx.html#close",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "close",
          "normalized": "InHandle-\u003eIO()",
          "package": "samtools",
          "signature": "InHandle-\u003eIO()",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#v:close"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eFetch a region specified by sequence name and coordinates, or the\n empty string when the sequence is not found.\n\u003c/p\u003e",
          "module": "Bio.SamTools.FaIdx",
          "name": "fetch",
          "package": "samtools",
          "signature": "InHandle-\u003e ByteString-\u003e (Int64, Int64)-\u003e IO ByteString",
          "type": "function"
        },
        "index": {
          "description": "Fetch region specified by sequence name and coordinates or the empty string when the sequence is not found",
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "fetch",
          "normalized": "InHandle-\u003eByteString-\u003e(Int,Int)-\u003eIO ByteString",
          "package": "samtools",
          "signature": "InHandle-\u003eByteString-\u003e(Int,Int)-\u003eIO ByteString",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#v:fetch"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.FaIdx",
          "name": "fetchLoc",
          "package": "samtools",
          "signature": "InHandle -\u003e OnSeq l -\u003e IO (Maybe ByteString)",
          "source": "src/Bio-SamTools-FaIdx.html#fetchLoc",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "fetchLoc",
          "normalized": "InHandle-\u003eOnSeq a-\u003eIO(Maybe ByteString)",
          "package": "samtools",
          "partial": "Loc",
          "signature": "InHandle-\u003eOnSeq l-\u003eIO(Maybe ByteString)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#v:fetchLoc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eName of the fasta file\n\u003c/p\u003e",
          "module": "Bio.SamTools.FaIdx",
          "name": "filename",
          "package": "samtools",
          "signature": "InHandle -\u003e FilePath",
          "source": "src/Bio-SamTools-FaIdx.html#filename",
          "type": "function"
        },
        "index": {
          "description": "Name of the fasta file",
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "filename",
          "normalized": "InHandle-\u003eFilePath",
          "package": "samtools",
          "signature": "InHandle-\u003eFilePath",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#v:filename"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "description": "\u003cp\u003eOpen an indexed fasta file\n\u003c/p\u003e",
          "module": "Bio.SamTools.FaIdx",
          "name": "open",
          "package": "samtools",
          "signature": "FilePath -\u003e IO InHandle",
          "source": "src/Bio-SamTools-FaIdx.html#open",
          "type": "function"
        },
        "index": {
          "description": "Open an indexed fasta file",
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "open",
          "normalized": "FilePath-\u003eIO InHandle",
          "package": "samtools",
          "signature": "FilePath-\u003eIO InHandle",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#v:open"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.FaIdx",
          "name": "readLoc",
          "package": "samtools",
          "signature": "FilePath -\u003e OnSeq l -\u003e IO (Maybe ByteString)",
          "source": "src/Bio-SamTools-FaIdx.html#readLoc",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "readLoc",
          "normalized": "FilePath-\u003eOnSeq a-\u003eIO(Maybe ByteString)",
          "package": "samtools",
          "partial": "Loc",
          "signature": "FilePath-\u003eOnSeq l-\u003eIO(Maybe ByteString)",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#v:readLoc"
      }
    },
    {
      "cmd": "insert",
      "document": {
        "description": {
          "module": "Bio.SamTools.FaIdx",
          "name": "withFastaIndex",
          "package": "samtools",
          "signature": "FilePath -\u003e (InHandle -\u003e IO a) -\u003e IO a",
          "source": "src/Bio-SamTools-FaIdx.html#withFastaIndex",
          "type": "function"
        },
        "index": {
          "hierarchy": "Bio SamTools FaIdx",
          "module": "Bio.SamTools.FaIdx",
          "name": "withFastaIndex",
          "normalized": "FilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "package": "samtools",
          "partial": "Fasta Index",
          "signature": "FilePath-\u003e(InHandle-\u003eIO a)-\u003eIO a",
          "type": "function"
        },
        "uri": "http://hackage.haskell.org/package/samtools/docs/Bio-SamTools-FaIdx.html#v:withFastaIndex"
      }
    }
  ]
]